<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xizhi Xu" />

<meta name="date" content="2024-10-21" />

<title>CAD cTWAS (no LD)</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CAD_cTWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xizhixu1016/CAD_cTWAS">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">CAD cTWAS (no LD)</h1>
<h4 class="author">Xizhi Xu</h4>
<h4 class="date">10/21/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CAD_cTWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxizhixu1016CADcTWAStree42784da268c96e6f1a3141dd93b8d47fcc14f2fetargetblank42784daa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xizhixu1016/CAD_cTWAS/tree/42784da268c96e6f1a3141dd93b8d47fcc14f2fe" target="_blank">42784da</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxizhixu1016CADcTWAStree42784da268c96e6f1a3141dd93b8d47fcc14f2fetargetblank42784daa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xizhixu1016/CAD_cTWAS/tree/42784da268c96e6f1a3141dd93b8d47fcc14f2fe" target="_blank">42784da</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  all_artery_weights.RData
    Untracked:  ctwas_res_all.RData
    Untracked:  ctwas_res_noLD.RData
    Untracked:  res.RData
    Untracked:  weights.RData
    Untracked:  weights_group.RData
    Untracked:  z_snp.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/CAD_cTWAS_noLD.Rmd</code>) and HTML (<code>docs/CAD_cTWAS_noLD.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/42784da268c96e6f1a3141dd93b8d47fcc14f2fe/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">42784da</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-25
</td>
<td>
Update CAD cTWAS with all tissues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/fe8c98f4a75032fb6cb7ece567b00104b9dbaec5/docs/CAD_cTWAS_noLD.html" target="_blank">fe8c98f</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/652b6f53e6d865c05aac8db27af3e787e9068347/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">652b6f5</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
<td>
Update CAD cTWAS with all tissues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/a35e278297c8873f422c14205a576f00697948bb/docs/CAD_cTWAS_noLD.html" target="_blank">a35e278</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/1d8cba974d4c672bb38adc1817bebd646d40c4d1/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">1d8cba9</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-15
</td>
<td>
Update CAD cTWAS
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/5b1f986a50470ba73ea2c2f212f0def4e63c3255/docs/CAD_cTWAS_noLD.html" target="_blank">5b1f986</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/77e1b208fb9e8d5120d2dd651a1bd073daa087b5/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">77e1b20</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
<td>
Update my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/2c5fce1d09b7be57bcff0ec4d3a966b3c8506e77/docs/CAD_cTWAS_noLD.html" target="_blank">2c5fce1</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/7eab91ecfb3ba7de8b792bf31f513c6d336ee811/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">7eab91e</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-04
</td>
<td>
Add my first analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><strong>Version Notes</strong></p>
<p>This is the no LD version of cTWAS. The prediction model includes tissues “Aorta”, “Coronary”, “Tibial” with their expression and splicing. The genome build of prediction model and reference was HG38.</p>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<p>Load the required packages.</p>
<pre class="r"><code>### Set up
# setwd(&quot;xizhi_xin_midway3/cTWAS_2024&quot;)
library(data.table)
library(ctwas)
library(RSQLite)
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(workflowr)
library(enrichR)</code></pre>
</div>
<div id="prepare-the-input-gwas-summary-statistics-reference-region-info-snp-info-prediction-model-weights" class="section level2">
<h2>Prepare the input: GWAS summary statistics, Reference (region info, SNP info), Prediction model (weights)</h2>
<div id="gwas-sum-stats" class="section level3">
<h3>GWAS Sum Stats</h3>
<p>The CAD summary statistics data I used was accessed from website: <a href="https://www.nature.com/articles/s41588-022-01233-6" class="uri">https://www.nature.com/articles/s41588-022-01233-6</a> . I filtered the original data to maintain the MAF between (0.01, 0.99). The sample size was 1165690.</p>
<pre class="r"><code>gwas_data &lt;- fread(&quot;GCST90132314_buildGRCh37.tsv&quot;, sep = &quot;\t&quot;)

# Filter to remain the variants with MAF (0.01, 0.99) 
gwas_filter &lt;- gwas_data[gwas_data$effect_allele_frequency &gt; 0.01, ]
gwas_filter &lt;- gwas_filter[gwas_filter$effect_allele_frequency &lt; 0.99, ]
gwas_filter$z &lt;- gwas_filter$beta / gwas_filter$standard_error

gwas_map &lt;- gwas_filter[, c(&quot;chromosome&quot;, &quot;base_pair_location&quot;, &quot;z&quot;)]

gwas_n &lt;- 1165690</code></pre>
</div>
<div id="reference" class="section level3">
<h3>Reference</h3>
<p>Here I got the region info from the package cTWAS. SNP info was also accessed from reference panel of cTWAS website. Another thing to notice: genome build needs to be matched. (Here: GRCh38 / hg38)</p>
<pre class="r"><code># Region info
region_file &lt;- system.file(&quot;extdata/ldetect&quot;, &quot;EUR.b38.ldetect.regions.RDS&quot;, package = &quot;ctwas&quot;)
region_info &lt;- readRDS(region_file)

# SNP info
ref_snp_info &lt;- fread(&quot;ukb_b38_0.1_var_info.Rvar&quot;, sep = &quot;\t&quot;) # 9309375
class(ref_snp_info) &lt;- &quot;data.frame&quot;

res &lt;- create_snp_map(region_info, ref_snp_info, ncore = 6)
region_info &lt;- res$region_info
snp_map &lt;- res$snp_map

head(region_info)

ref_data &lt;- read.table(&quot;ukb_b37_0.1_var_info.Rvar&quot;, header = TRUE, sep = &quot;\t&quot;)

colnames(gwas_map)[colnames(gwas_map) == &quot;base_pair_location&quot;] &lt;- &quot;pos&quot;
colnames(gwas_map)[colnames(gwas_map) == &quot;chromosome&quot;] &lt;- &quot;chrom&quot;
head(gwas_map)

# Merge to find rsID based on positions
z_snp &lt;- merge(gwas_map, ref_data, by = c(&quot;pos&quot;, &quot;chrom&quot;)) # 9155485
z_snp &lt;- z_snp[, c(&quot;id&quot;, &quot;alt&quot;, &quot;ref&quot;, &quot;z&quot;)]
colnames(z_snp)[colnames(z_snp) == &quot;alt&quot;] &lt;- &quot;A1&quot;
colnames(z_snp)[colnames(z_snp) == &quot;ref&quot;] &lt;- &quot;A2&quot;
head(z_snp)</code></pre>
<p>Harmonize GWAS summary statistics:</p>
<pre class="r"><code># Harmonize GWAS summary statistics
load(&quot;z_snp.RData&quot;)
load(&quot;res.RData&quot;)
region_info &lt;- res$region_info
snp_map &lt;- res$snp_map
z_snp &lt;- preprocess_z_snp(z_snp, snp_map)</code></pre>
<pre><code>2024-10-25 11:51:06 INFO::Preprocessing z_snp...
2024-10-25 11:51:16 INFO::z_snp has 9155485 variants in total
2024-10-25 11:51:20 INFO::9142078 variants left after filtering by the reference SNPs.
2024-10-25 11:51:21 INFO::Remove 14480 multiallelic variants
2024-10-25 11:51:26 INFO::Harmonize 9127598 variants between GWAS and the reference
2024-10-25 11:51:42 INFO::Flip signs for 1272047 (13.94%) variants
2024-10-25 11:51:42 INFO::Remove 1272047 strand ambiguous variants
2024-10-25 11:51:42 INFO::7855551 variants left after preprocessing and harmonization.</code></pre>
<pre class="r"><code>dim(z_snp)</code></pre>
<pre><code>[1] 7855551       4</code></pre>
</div>
<div id="prediction-model" class="section level3">
<h3>Prediction model</h3>
<p>The prediction model was accessed from GTEx. The molecular traits were expression and splicing. Here, I included all tissues related to artery: Aorta, Coronary, Tibial.</p>
<p>The number of each group:</p>
<pre class="r"><code>load(&quot;weights_group.RData&quot;)
print(weights_group)</code></pre>
<pre><code>   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
              10403               29657                9496               29290 
  Tibial|expression     Tibial|splicing                 SNP 
              10600               30075             7851620 </code></pre>
<pre class="r"><code># Artery_Aorta
artery_aorta_eqtl &lt;- &quot;mashr_eqtl/eqtl/mashr/mashr_Artery_Aorta.db&quot;
artery_aorta_sqtl &lt;- &quot;mashr_sqtl/sqtl/mashr/mashr_Artery_Aorta.db&quot;

artery_aorta_eqtl_weights &lt;- preprocess_weights(artery_aorta_eqtl,
                                    region_info,
                                    z_snp$id,
                                    snp_map,
                                    type = &quot;expression&quot;,
                                    context = &quot;Aorta&quot;)

artery_aorta_sqtl_weights &lt;- preprocess_weights(artery_aorta_sqtl,
                                          region_info,
                                          z_snp$id,
                                          snp_map,
                                          type = &quot;splicing&quot;,
                                          context = &quot;Aorta&quot;)


# Artery_Coronary
artery_coronary_eqtl &lt;- &quot;mashr_eqtl/eqtl/mashr/mashr_Artery_Coronary.db&quot;
artery_coronary_sqtl &lt;- &quot;mashr_sqtl/sqtl/mashr/mashr_Artery_Coronary.db&quot;

artery_coronary_eqtl_weights &lt;- preprocess_weights(artery_coronary_eqtl,
                                                region_info,
                                                z_snp$id,
                                                snp_map,
                                                type = &quot;expression&quot;,
                                                context = &quot;Coronary&quot;)

artery_coronary_sqtl_weights &lt;- preprocess_weights(artery_coronary_sqtl,
                                          region_info,
                                          z_snp$id,
                                          snp_map,
                                          type = &quot;splicing&quot;,
                                          context = &quot;Coronary&quot;)

# Artery_Tibial
artery_tibial_eqtl &lt;- &quot;mashr_eqtl/eqtl/mashr/mashr_Artery_Tibial.db&quot;
artery_tibial_eqtl_weights &lt;- preprocess_weights(artery_tibial_eqtl,
                                                   region_info,
                                                   z_snp$id,
                                                   snp_map,
                                                   type = &quot;expression&quot;,
                                                   context = &quot;Tibial&quot;)

artery_tibial_sqtl &lt;- &quot;mashr_sqtl/sqtl/mashr/mashr_Artery_Tibial.db&quot;
artery_tibial_sqtl_weights &lt;- preprocess_weights(artery_tibial_sqtl,
                                                 region_info,
                                                 z_snp$id,
                                                 snp_map,
                                                 type = &quot;splicing&quot;,
                                                 context = &quot;Tibial&quot;)

weights &lt;- c(artery_aorta_eqtl_weights, artery_aorta_sqtl_weights, artery_coronary_eqtl_weights, 
             artery_coronary_sqtl_weights, artery_tibial_eqtl_weights, artery_tibial_sqtl_weights)</code></pre>
</div>
</div>
<div id="run-ctwas" class="section level2">
<h2>Run cTWAS</h2>
<p>After all the input files ready, run the cTWAS.</p>
<pre class="r"><code># # ctwas_res &lt;- ctwas_sumstats_noLD(z_snp,
#                                  artery,
#                                  region_info,
#                                  snp_map,
#                                  thin = 0.1, 
#                                  ncore = 6)


load(&quot;all_artery_weights.RData&quot;)
load(&quot;ctwas_res_all.RData&quot;)

z_gene &lt;- ctwas_res$z_gene
param &lt;- ctwas_res$param
finemap_res &lt;- ctwas_res$finemap_res
susie_alpha_res &lt;- ctwas_res$susie_alpha_res
boundary_genes &lt;- ctwas_res$boundary_genes
region_data &lt;- ctwas_res$region_data
screen_res &lt;- ctwas_res$screen_res</code></pre>
<div id="explore-and-interpret-the-results" class="section level3">
<h3>Explore and interpret the results</h3>
<p><strong>Create the convergence plots:</strong></p>
<pre class="r"><code>gwas_n &lt;- 1165690
make_convergence_plots(param, gwas_n)</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/fe8c98f4a75032fb6cb7ece567b00104b9dbaec5/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-7-1.png" target="_blank">fe8c98f</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/a35e278297c8873f422c14205a576f00697948bb/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-7-1.png" target="_blank">a35e278</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/5b1f986a50470ba73ea2c2f212f0def4e63c3255/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-7-1.png" target="_blank">5b1f986</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These plots showed the estimated prior inclusion probability, prior effect size variance, enrichment and proportion of variance explained (PVE) over the iterations of parameter estimation.</p>
<p><strong>Estimate the parameters:</strong></p>
<pre class="r"><code>ctwas_parameters &lt;- summarize_param(param, gwas_n)
ctwas_parameters</code></pre>
<pre><code>$group_size
   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
              10403               29657                9496               29290 
  Tibial|expression     Tibial|splicing                 SNP 
              10600               30075             7851620 

$group_prior
   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
       0.0078530848        0.0018636328        0.0015952678        0.0041753694 
  Tibial|expression     Tibial|splicing                 SNP 
       0.0056263115        0.0022117083        0.0001886633 

$group_prior_var
   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
           36.70085            12.33802            36.70085            12.33802 
  Tibial|expression     Tibial|splicing                 SNP 
           36.70085            12.33802            10.26662 

$enrichment
   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
          41.624862            9.878087            8.455633           22.131325 
  Tibial|expression     Tibial|splicing 
          29.821968           11.723043 

$group_pve
   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
       0.0025721242        0.0005849922        0.0004769439        0.0012944247 
  Tibial|expression     Tibial|splicing                 SNP 
       0.0018776848        0.0007040378        0.0130464165 

$total_pve
[1] 0.02055662

$prop_heritability
   Aorta|expression      Aorta|splicing Coronary|expression   Coronary|splicing 
         0.12512386          0.02845760          0.02320147          0.06296874 
  Tibial|expression     Tibial|splicing                 SNP 
         0.09134208          0.03424871          0.63465754 </code></pre>
<p><strong>Create the pie chart to see the heritability:</strong></p>
<p>ctwas_parameters$prop_heritability contained the proportion of heritability mediated by molecular traits and variants, we visualized it using pie chart.</p>
<pre class="r"><code>data &lt;- data.frame(
  category = names(ctwas_parameters$prop_heritability),
  percentage = ctwas_parameters$prop_heritability
)

data$percentage_label &lt;- paste0(round(data$percentage * 100, 1), &quot;%&quot;)

ggplot(data, aes(x = &quot;&quot;, y = percentage, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) +
  coord_polar(&quot;y&quot;, start = 0) +
  theme_void() +  # Remove background and axes
  geom_text(aes(label = percentage_label), 
            position = position_stack(vjust = 0.5), size = 5) +
  scale_fill_manual(values = c(&quot;#FF9999&quot;, &quot;#66B2FF&quot;, &quot;#99FF99&quot;, &quot;#FFCC99&quot;, &quot;#CCCCFF&quot;, &quot;#99CCCC&quot;, &quot;#FF6666&quot;)) + 
  labs(fill = &quot;Category&quot;) +
  ggtitle(&quot;Percent of heritability&quot;)</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/fe8c98f4a75032fb6cb7ece567b00104b9dbaec5/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-9-1.png" target="_blank">fe8c98f</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/a35e278297c8873f422c14205a576f00697948bb/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-9-1.png" target="_blank">a35e278</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Create a diagnostic plot:</strong></p>
<p>For all genes analyzed, we compared the z-scores and fine-mapping PIPs. We generally expect high PIP molecular traits to have high Z-scores as well.</p>
<pre class="r"><code>ggplot(data = finemap_res[finemap_res$type!=&quot;SNP&quot;,], aes(x = abs(z), y = susie_pip)) +
  geom_point() +
  labs(x = &quot;abs(z-scores)&quot;, y = &quot;PIPs&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/fe8c98f4a75032fb6cb7ece567b00104b9dbaec5/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-10-1.png" target="_blank">fe8c98f</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="inspect-and-summarie-results" class="section level3">
<h3>Inspect and summarie results</h3>
<p><strong>Add p-values to finemapping results:</strong></p>
<pre class="r"><code>finemap_res$pval &lt;- z2p(finemap_res$z)
head(finemap_res)</code></pre>
<pre><code>                                   id       molecular_id       type context
1 ENSG00000162585.16|Aorta_expression ENSG00000162585.16 expression   Aorta
2 ENSG00000116151.13|Aorta_expression ENSG00000116151.13 expression   Aorta
3 ENSG00000157916.19|Aorta_expression ENSG00000157916.19 expression   Aorta
4  ENSG00000157911.9|Aorta_expression  ENSG00000157911.9 expression   Aorta
5 ENSG00000149527.17|Aorta_expression ENSG00000149527.17 expression   Aorta
6 ENSG00000157873.17|Aorta_expression ENSG00000157873.17 expression   Aorta
             group         region_id           z    susie_pip       mu2   cs
1 Aorta|expression 1_1961168_3666172 -1.87356418 1.930207e-09 4.2999722 &lt;NA&gt;
2 Aorta|expression 1_1961168_3666172 -1.29114760 7.870070e-10 2.5532741 &lt;NA&gt;
3 Aorta|expression 1_1961168_3666172  2.22492952 3.890599e-09 5.6646595 &lt;NA&gt;
4 Aorta|expression 1_1961168_3666172 -0.68014094 4.378936e-10 1.4118524 &lt;NA&gt;
5 Aorta|expression 1_1961168_3666172 -0.29444335 3.646776e-10 1.0556341 &lt;NA&gt;
6 Aorta|expression 1_1961168_3666172  0.09912788 3.512851e-10 0.9827874 &lt;NA&gt;
        pval
1 0.06099052
2 0.19665250
3 0.02608597
4 0.49641522
5 0.76841914
6 0.92103674</code></pre>
<p><strong>Add gene annotations:(make sure the genome build match)</strong></p>
<p>Then we mapped the molecular traits to their corresponding genes to compute gene PIPs.</p>
<pre class="r"><code>ens_db &lt;- EnsDb.Hsapiens.v86
mapping_table &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS&quot;)</code></pre>
<p><strong>Update finemapping results with gene annotation:</strong></p>
<pre class="r"><code># Update finemap results with gene annot
finemap_res &lt;- anno_finemap_res(finemap_res,
                                snp_map = snp_map,
                                mapping_table = mapping_table,
                                add_gene_annot = TRUE,
                                map_by = &quot;molecular_id&quot;,
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-25 11:51:53 INFO::Annotating fine-mapping result ...
2024-10-25 11:51:53 INFO::Map molecular traits to genes
2024-10-25 11:51:53 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-25 11:51:56 INFO::Add gene positions
2024-10-25 11:51:59 INFO::Add SNP positions</code></pre>
<pre class="r"><code>finemap_res_sig &lt;- subset(finemap_res, group != &quot;SNP&quot; &amp; gene_type == &quot;protein_coding&quot; &amp; susie_pip &gt; 0.8 &amp; !is.na(cs))</code></pre>
<p>Compute gene PIPs across different types of molecular traits. Here, “group_by = gene_name”, because molecular traits are mapped to their corresponding genes by the “gene_name” column.</p>
<pre class="r"><code>susie_alpha_res &lt;- anno_susie_alpha_res(susie_alpha_res,
                                        mapping_table = mapping_table,
                                        map_by = &quot;molecular_id&quot;,
                                        drop_unmapped = TRUE)</code></pre>
<pre><code>2024-10-25 11:52:01 INFO::Annotating susie alpha result ...
2024-10-25 11:52:01 INFO::Map molecular traits to genes
2024-10-25 11:52:01 INFO::Split PIPs for molecular traits mapped to multiple genes</code></pre>
<pre class="r"><code>combined_pip_by_type &lt;- combine_gene_pips(susie_alpha_res,
                                          group_by = &quot;gene_name&quot;,
                                          by = &quot;type&quot;,
                                          method = &quot;combine_cs&quot;,
                                          filter_cs = TRUE,
                                          include_cs_id = TRUE)

combined_pip_by_type_sig &lt;- subset(combined_pip_by_type, combined_pip &gt; 0.8)

DT::datatable(combined_pip_by_type_sig,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Gene PIPs, only genes with credible sets are shown&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-232a8e63d6aacd296363" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-232a8e63d6aacd296363">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Gene PIPs, only genes with credible sets are shown<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36"],["TRMT61B","TBC1D7","ZNHIT3","PHACTR1","BOP1","NBEAL1","TCF21","IP6K2","ITIH4","PTK7","ITGA1","TBXAS1","SAP130","CCM2","MRAS","LIPA","TGFB1","PDLIM5","FES","CETP","CDK8","MAP1S","BMP1","CORO6","ARVCF","ZSCAN31","GSTT2B","SERPINH1","MIA3","SIPA1","R3HDM2","EIF2B2","RRBP1","C6orf106","PSMA3-AS1","CALCRL"],["2_28375910_28994693.L1","6_13209156_14802693.L1","17_36141651_38653091.L1","6_11790583_13209156.L1","8_143155464_145078481.L1","2_201953914_204934518.L1","6_132444530_133923105.L1","3_47685722_49279539.L1","3_51797999_54047363.L1","6_42070983_43788432.L1","5_52838788_54407214.L1","7_139059933_140233377.L1","2_127276771_128716867.L1","7_44724229_45913323.L1","3_137652241_140235755.L1","10_87367307_89253624.L1","19_40478694_41625199.L1","4_93312148_95261037.L1","15_89932319_91621162.L1","16_55869862_57630418.L1","13_25210224_26710225.L1","19_16263605_18299052.L1","8_21804225_23039544.L1","17_29007226_31459473.L1","22_19924835_22002927.L1","6_28050041_28949831.L1","22_23370460_24588236.L1","11_74701306_77086163.L1","1_222056878_224750818.L1","11_64037097_66131160.L1","12_57155077_58914885.L1","14_72422907_75978424.L1","20_17506587_18991559.L1","6_33268720_35487979.L1","14_57014544_58981618.L1","2_184415446_189017339.L1"],[1,0.99999999999859,0.9999999986804733,0.9999999986665887,0.9999977207557993,0.9977419225293662,0.9949814265971395,0.9929120590689428,0.9911879286097685,0.9840105982456073,0.9829025810527523,0.982881331447818,0.9793128705289912,0.9758817472749568,0.9729755965556406,0.9661834465598135,0.9603705130413827,0.9571234468497776,0.9567800000313702,0.9530128976621716,0.9260935181543878,0.9260536033551483,0.9214673199012092,0.9057661566689741,0.8993058209690633,0.895432078569548,0.8896169670592035,0.8896100605462344,0.8874382897006723,0.8867861046286577,0.881179888062415,0.8699079096379659,0.8591130455347936,0.8533544401385483,0.8038404689519556,0.8019895317733428],[null,"6_13209156_14802693.L1","17_36141651_38653091.L1","6_11790583_13209156.L1","8_143155464_145078481.L1","2_201953914_204934518.L1","6_132444530_133923105.L1",null,null,"6_42070983_43788432.L1",null,null,"2_127276771_128716867.L1",null,"3_137652241_140235755.L1","10_87367307_89253624.L1","19_40478694_41625199.L1",null,"15_89932319_91621162.L1","16_55869862_57630418.L1","13_25210224_26710225.L1","19_16263605_18299052.L1","8_21804225_23039544.L1","17_29007226_31459473.L1","22_19924835_22002927.L1",null,"22_23370460_24588236.L1","11_74701306_77086163.L1",null,null,null,"14_72422907_75978424.L1",null,"6_33268720_35487979.L1",null,"2_184415446_189017339.L1"],[null,0.99999999999859,0.9999999986804733,0.9999999986665887,0.9999977207557993,0.9977419225293662,0.9949814265971395,null,null,0.9840105982456073,null,null,0.9793128705289912,null,0.9729755965556406,0.929975873549565,0.9603705130413827,null,0.9567800000313702,0.9530128976621716,0.9260935181543878,0.7949466545016703,0.4335990366357124,0.4503065441917729,0.6548101610864523,null,0.8896169670592035,0.8896100605462344,null,null,null,0.8608294314926849,null,0.4284729044785814,null,0.7928952550351068],["2_28375910_28994693.L1",null,null,null,null,null,null,"3_47685722_49279539.L1","3_51797999_54047363.L1",null,"5_52838788_54407214.L1","7_139059933_140233377.L1",null,"7_44724229_45913323.L1",null,"10_87367307_89253624.L1",null,"4_93312148_95261037.L1",null,null,null,"19_16263605_18299052.L1","8_21804225_23039544.L1","17_29007226_31459473.L1","22_19924835_22002927.L1","6_28050041_28949831.L1",null,null,"1_222056878_224750818.L1","11_64037097_66131160.L1","12_57155077_58914885.L1","14_72422907_75978424.L1","20_17506587_18991559.L1","6_33268720_35487979.L1","14_57014544_58981618.L1","2_184415446_189017339.L1"],[1,null,null,null,null,null,null,0.9929120590689428,0.9911879286097685,null,0.9829025810527523,0.982881331447818,null,0.9758817472749568,null,0.03620757301024846,null,0.9571234468497776,null,null,null,0.131106948853478,0.4878682832654968,0.4554596124772012,0.2444956598826109,0.895432078569548,null,null,0.8874382897006723,0.8867861046286577,0.881179888062415,0.009078478145281066,0.8591130455347936,0.4248815356599669,0.8038404689519556,0.009094276738235929]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_cs_id<\/th>\n      <th>combined_pip<\/th>\n      <th>expression_cs_id<\/th>\n      <th>expression_pip<\/th>\n      <th>splicing_cs_id<\/th>\n      <th>splicing_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualize-by-the-gene-locus-plots" class="section level3">
<h3>Visualize by the gene locus plots</h3>
<p>We created locus plots for the top 3 gene regions.</p>
<pre class="r"><code>make_locusplot(finemap_res,
               region_id = &quot;2_28375910_28994693&quot;,
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;)</code></pre>
<pre><code>2024-10-25 11:52:03 INFO::Limit to protein coding genes
2024-10-25 11:52:03 INFO::focal id: intron_2_28865119_28869560|Coronary_splicing
2024-10-25 11:52:03 INFO::focal molecular trait: TRMT61B Coronary splicing
2024-10-25 11:52:03 INFO::Range of locus: chr2:28375859-29009092</code></pre>
<pre><code>chromosome 2, position 28375859 to 29009092</code></pre>
<pre><code>1811 SNPs/datapoints</code></pre>
<pre><code>2024-10-25 11:52:05 INFO::focal molecular trait QTL positions: 28865472,28865584
2024-10-25 11:52:05 INFO::Limit PIPs to credible sets</code></pre>
<pre><code>Warning: ggrepel: 76 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/a35e278297c8873f422c14205a576f00697948bb/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-15-1.png" target="_blank">a35e278</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/5b1f986a50470ba73ea2c2f212f0def4e63c3255/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-15-1.png" target="_blank">5b1f986</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>make_locusplot(finemap_res,
               region_id = &quot;6_13209156_14802693&quot;,
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;)</code></pre>
<pre><code>2024-10-25 11:52:06 INFO::Limit to protein coding genes
2024-10-25 11:52:06 INFO::focal id: ENSG00000145979.17|Tibial_expression
2024-10-25 11:52:06 INFO::focal molecular trait: TBC1D7 Tibial expression
2024-10-25 11:52:06 INFO::Range of locus: chr6:13209417-14802556</code></pre>
<pre><code>chromosome 6, position 13209417 to 14802556</code></pre>
<pre><code>4389 SNPs/datapoints</code></pre>
<pre><code>2024-10-25 11:52:06 INFO::focal molecular trait QTL positions: 13267390,13329920
2024-10-25 11:52:06 INFO::Limit PIPs to credible sets</code></pre>
<pre><code>Warning: ggrepel: 77 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>make_locusplot(finemap_res,
               region_id = &quot;17_36141651_38653091&quot;,
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;)</code></pre>
<pre><code>2024-10-25 11:52:07 INFO::Limit to protein coding genes
2024-10-25 11:52:07 INFO::focal id: ENSG00000273611.4|Aorta_expression
2024-10-25 11:52:07 INFO::focal molecular trait: ZNHIT3 Aorta expression
2024-10-25 11:52:07 INFO::Range of locus: chr17:36143579-38652996</code></pre>
<pre><code>chromosome 17, position 36143579 to 38652996</code></pre>
<pre><code>4426 SNPs/datapoints</code></pre>
<pre><code>2024-10-25 11:52:07 INFO::focal molecular trait QTL positions: 36486677,36488913
2024-10-25 11:52:07 INFO::Limit PIPs to credible sets</code></pre>
<pre><code>Warning: ggrepel: 165 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>The top track showed -log10(p-value) of the association of variants (from CAD GWAS) and molecular traits (from the package computed z-scores) with the phenotype.</p>
<p>The mid track showed PIPs of variants and molecular traits. We limited PIP results to credible sets in the PIP track (filter_cs = TRUE).</p>
<p>The next track showed the QTLs of the focal gene. We chose the molecular trait with the top 3 highest PIP (TRMT61B, TBC1D7, ZNHIT3).</p>
<p>The bottom was the gene track. We limited results to protein coding genes by default. We could draw a red dotted line to show the PIP cutoff of 0.8 by setting highlight_pip = 0.8.</p>
<p>As conclusion, we found TRMT61B, TBC1D7, ZNHIT3 as the likely causal genes of CAD. Previously, we only included Aorta tissue in the prediction model and we found ZNHIT3 as likely causal gene. Now we added 2 another tissues and we got more genes, ZNHIT3 was still there, which consistent with our previous findings.</p>
</div>
<div id="gene-set-enrichment-analysis" class="section level3">
<h3>Gene set enrichment analysis</h3>
<p>At last, we did enrichment analysis using the genes with PIP &gt; 0.8.</p>
<pre class="r"><code>dbs &lt;- c(&quot;GO_Biological_Process_2023&quot;, &quot;GO_Cellular_Component_2023&quot;, &quot;GO_Molecular_Function_2023&quot;)
genes &lt;- combined_pip_by_type$gene_name[combined_pip_by_type$combined_pip &gt; 0.8]

# number of genes for gene set enrichment
sprintf(&quot;The number of genes used in enrichment analysis = %s&quot;, length(genes))</code></pre>
<pre><code>[1] &quot;The number of genes used in enrichment analysis = 36&quot;</code></pre>
<pre class="r"><code>GO_enrichment &lt;- enrichr(genes, dbs)</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
  Querying GO_Cellular_Component_2023... Done.
  Querying GO_Molecular_Function_2023... Done.
Parsing results... Done.</code></pre>
<p>Biological process: we found one GO term “Circulatory System Development”.</p>
<pre class="r"><code>db1 &lt;- &quot;GO_Biological_Process_2023&quot;
df1 &lt;- GO_enrichment[[db1]]
print(plotEnrich(GO_enrichment[[db1]]))</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/fe8c98f4a75032fb6cb7ece567b00104b9dbaec5/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-17-1.png" target="_blank">fe8c98f</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df1 &lt;- df1[df1$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]
print(df1)</code></pre>
<pre><code>                                         Term Overlap Adjusted.P.value
1 Circulatory System Development (GO:0072359)   4/132       0.04913197
                    Genes
1 TGFB1;TCF21;CCM2;PDLIM5</code></pre>
<p>Cellular component: we didn’t find any GO terms.</p>
<pre class="r"><code>db2 &lt;- &quot;GO_Cellular_Component_2023&quot;
df2 &lt;- GO_enrichment[[db2]]
print(plotEnrich(GO_enrichment[[db2]]))</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df2 &lt;- df2[df2$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]</code></pre>
<p>Molecular function: we didn’t find any GO terms.</p>
<pre class="r"><code>db3 &lt;- &quot;GO_Molecular_Function_2023&quot;
df3 &lt;- GO_enrichment[[db3]]
print(plotEnrich(GO_enrichment[[db3]]))</code></pre>
<pre><code>Warning in plotEnrich(GO_enrichment[[db3]]): There are duplicated trimmed names
in the plot, consider increasing the &#39;numChar&#39; setting.</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/fe8c98f4a75032fb6cb7ece567b00104b9dbaec5/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-19-1.png" target="_blank">fe8c98f</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df3 &lt;- df3[df3$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]</code></pre>
<p>To sum up, Circulatory System Development refers to a biological process that involves the formation and maturation of the circulatory system, which includes the heart, blood vessels, and the blood itself. Since my research topic, CAD, is closely related to abnormalities in the development and function of the circulatory system, including processes like blood vessel formation and heart development, this result makes sense.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux 8.4 (Ootpa)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el8-x86_64/lib/libopenblas_skylakexp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] enrichR_3.2               ggplot2_3.4.3            
 [3] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.22.0         
 [5] AnnotationFilter_1.22.0   GenomicFeatures_1.50.4   
 [7] AnnotationDbi_1.60.2      Biobase_2.58.0           
 [9] GenomicRanges_1.50.2      GenomeInfoDb_1.34.9      
[11] IRanges_2.32.0            S4Vectors_0.36.2         
[13] BiocGenerics_0.44.0       RSQLite_2.3.1            
[15] ctwas_0.4.15              data.table_1.14.8        
[17] workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.38.0              locuszoomr_0.3.5           
  [7] fs_1.5.2                    rstudioapi_0.14            
  [9] farver_2.1.1                DT_0.22                    
 [11] ggrepel_0.9.3               bit64_4.0.5                
 [13] fansi_1.0.4                 xml2_1.3.3                 
 [15] codetools_0.2-18            logging_0.10-108           
 [17] cachem_1.0.8                knitr_1.42                 
 [19] jsonlite_1.8.7              Rsamtools_2.14.0           
 [21] dbplyr_2.3.2                png_0.1-7                  
 [23] readr_2.1.4                 compiler_4.2.0             
 [25] httr_1.4.7                  Matrix_1.6-4               
 [27] fastmap_1.1.1               lazyeval_0.2.2             
 [29] cli_3.6.2                   later_1.3.0                
 [31] htmltools_0.5.7             prettyunits_1.1.1          
 [33] tools_4.2.0                 gtable_0.3.4               
 [35] glue_1.6.2                  GenomeInfoDbData_1.2.9     
 [37] dplyr_1.1.3                 rappdirs_0.3.3             
 [39] Rcpp_1.0.11                 jquerylib_0.1.4            
 [41] vctrs_0.6.3                 Biostrings_2.66.0          
 [43] rtracklayer_1.58.0          crosstalk_1.2.0            
 [45] xfun_0.40                   stringr_1.5.0              
 [47] ps_1.7.0                    lifecycle_1.0.4            
 [49] irlba_2.3.5                 restfulr_0.0.15            
 [51] WriteXLS_6.7.0              XML_3.99-0.9               
 [53] getPass_0.2-2               zlibbioc_1.44.0            
 [55] zoo_1.8-11                  scales_1.2.1               
 [57] gggrid_0.2-0                hms_1.1.3                  
 [59] promises_1.2.0.1            MatrixGenerics_1.10.0      
 [61] ProtGenerics_1.30.0         parallel_4.2.0             
 [63] SummarizedExperiment_1.28.0 LDlinkR_1.3.0              
 [65] yaml_2.3.7                  curl_4.3.2                 
 [67] memoise_2.0.1               sass_0.4.1                 
 [69] biomaRt_2.54.1              stringi_1.7.12             
 [71] highr_0.9                   BiocIO_1.8.0               
 [73] filelock_1.0.2              BiocParallel_1.32.6        
 [75] rlang_1.1.1                 pkgconfig_2.0.3            
 [77] matrixStats_1.2.0           bitops_1.0-7               
 [79] evaluate_0.15               lattice_0.20-45            
 [81] purrr_1.0.2                 labeling_0.4.3             
 [83] GenomicAlignments_1.34.1    htmlwidgets_1.6.2          
 [85] cowplot_1.1.1               bit_4.0.5                  
 [87] processx_3.5.3              tidyselect_1.2.0           
 [89] magrittr_2.0.3              R6_2.5.1                   
 [91] generics_0.1.3              DelayedArray_0.24.0        
 [93] DBI_1.1.3                   withr_2.5.0                
 [95] pgenlibr_0.3.6              pillar_1.9.0               
 [97] whisker_0.4                 KEGGREST_1.38.0            
 [99] RCurl_1.98-1.12             mixsqp_0.3-54              
[101] tibble_3.2.1                crayon_1.5.2               
[103] utf8_1.2.3                  BiocFileCache_2.6.1        
[105] plotly_4.10.0               tzdb_0.4.0                 
[107] rmarkdown_2.21              progress_1.2.2             
[109] grid_4.2.0                  blob_1.2.4                 
[111] callr_3.7.0                 git2r_0.30.1               
[113] digest_0.6.33               tidyr_1.3.0                
[115] httpuv_1.6.5                munsell_0.5.0              
[117] viridisLite_0.4.2           bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
