<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xizhi Xu" />

<meta name="date" content="2024-10-14" />

<title>CAD cTWAS (no LD)</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CAD_cTWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xizhixu1016/CAD_cTWAS">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">CAD cTWAS (no LD)</h1>
<h4 class="author">Xizhi Xu</h4>
<h4 class="date">10/14/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CAD_cTWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxizhixu1016CADcTWAStree1d8cba974d4c672bb38adc1817bebd646d40c4d1targetblank1d8cba9a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xizhixu1016/CAD_cTWAS/tree/1d8cba974d4c672bb38adc1817bebd646d40c4d1" target="_blank">1d8cba9</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxizhixu1016CADcTWAStree1d8cba974d4c672bb38adc1817bebd646d40c4d1targetblank1d8cba9a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xizhixu1016/CAD_cTWAS/tree/1d8cba974d4c672bb38adc1817bebd646d40c4d1" target="_blank">1d8cba9</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  ctwas_res_noLD.RData
    Untracked:  res.RData
    Untracked:  weights.RData
    Untracked:  z_snp.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/CAD_cTWAS_noLD.Rmd</code>) and HTML (<code>docs/CAD_cTWAS_noLD.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/1d8cba974d4c672bb38adc1817bebd646d40c4d1/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">1d8cba9</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-15
</td>
<td>
Update CAD cTWAS
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/5b1f986a50470ba73ea2c2f212f0def4e63c3255/docs/CAD_cTWAS_noLD.html" target="_blank">5b1f986</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/77e1b208fb9e8d5120d2dd651a1bd073daa087b5/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">77e1b20</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
<td>
Update my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/2c5fce1d09b7be57bcff0ec4d3a966b3c8506e77/docs/CAD_cTWAS_noLD.html" target="_blank">2c5fce1</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/7eab91ecfb3ba7de8b792bf31f513c6d336ee811/analysis/CAD_cTWAS_noLD.Rmd" target="_blank">7eab91e</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-04
</td>
<td>
Add my first analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<p>Load the required packages.</p>
<pre class="r"><code>### Set up
# setwd(&quot;xizhi_xin_midway3/cTWAS_2024&quot;)
library(data.table)
library(ctwas)
library(RSQLite)
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(workflowr)</code></pre>
</div>
<div id="prepare-the-input-gwas-summary-statistics-reference-region-info-snp-info-prediction-model-weights" class="section level2">
<h2>Prepare the input: GWAS summary statistics, Reference (region info, SNP info), Prediction model (weights)</h2>
<div id="gwas-sum-stats" class="section level3">
<h3>GWAS Sum Stats</h3>
<p>The CAD summary statistics data I used was accessed from website: <a href="https://www.nature.com/articles/s41588-022-01233-6" class="uri">https://www.nature.com/articles/s41588-022-01233-6</a> . I filtered the original data to maintain the MAF between (0.01, 0.99).</p>
<pre class="r"><code>gwas_data &lt;- fread(&quot;GCST90132314_buildGRCh37.tsv&quot;, sep = &quot;\t&quot;)

# Filter to remain the variants with MAF (0.01, 0.99) 
gwas_filter &lt;- gwas_data[gwas_data$effect_allele_frequency &gt; 0.01, ]
gwas_filter &lt;- gwas_filter[gwas_filter$effect_allele_frequency &lt; 0.99, ]
gwas_filter$z &lt;- gwas_filter$beta / gwas_filter$standard_error

gwas_map &lt;- gwas_filter[, c(&quot;chromosome&quot;, &quot;base_pair_location&quot;, &quot;z&quot;)]</code></pre>
</div>
<div id="reference" class="section level3">
<h3>Reference</h3>
<p>Here I got the region info from the package cTWAS. SNP info was also accessed from reference panel of cTWAS website. Another thing to notice: genome build needs to be matched. (Here: GRCh38 / hg38)</p>
<pre class="r"><code># Region info
region_file &lt;- system.file(&quot;extdata/ldetect&quot;, &quot;EUR.b38.ldetect.regions.RDS&quot;, package = &quot;ctwas&quot;)
region_info &lt;- readRDS(region_file)

# SNP info
ref_snp_info &lt;- fread(&quot;ukb_b38_0.1_var_info.Rvar&quot;, sep = &quot;\t&quot;) # 9309375
class(ref_snp_info) &lt;- &quot;data.frame&quot;

res &lt;- create_snp_map(region_info, ref_snp_info, ncore = 6)
region_info &lt;- res$region_info
snp_map &lt;- res$snp_map

head(region_info)

ref_data &lt;- read.table(&quot;ukb_b37_0.1_var_info.Rvar&quot;, header = TRUE, sep = &quot;\t&quot;)

colnames(gwas_map)[colnames(gwas_map) == &quot;base_pair_location&quot;] &lt;- &quot;pos&quot;
colnames(gwas_map)[colnames(gwas_map) == &quot;chromosome&quot;] &lt;- &quot;chrom&quot;
head(gwas_map)

# Merge to find rsID based on positions
z_snp &lt;- merge(gwas_map, ref_data, by = c(&quot;pos&quot;, &quot;chrom&quot;)) # 9155485
z_snp &lt;- z_snp[, c(&quot;id&quot;, &quot;alt&quot;, &quot;ref&quot;, &quot;z&quot;)]
colnames(z_snp)[colnames(z_snp) == &quot;alt&quot;] &lt;- &quot;A1&quot;
colnames(z_snp)[colnames(z_snp) == &quot;ref&quot;] &lt;- &quot;A2&quot;
head(z_snp)</code></pre>
<p>Harmonize GWAS summary statistics:</p>
<pre class="r"><code># Harmonize GWAS summary statistics
load(&quot;z_snp.RData&quot;)
load(&quot;res.RData&quot;)
region_info &lt;- res$region_info
snp_map &lt;- res$snp_map
z_snp &lt;- preprocess_z_snp(z_snp, snp_map)</code></pre>
<pre><code>2024-10-15 12:33:20 INFO::Preprocessing z_snp...
2024-10-15 12:33:30 INFO::z_snp has 9155485 variants in total
2024-10-15 12:33:35 INFO::9142078 variants left after filtering by the reference SNPs.
2024-10-15 12:33:36 INFO::Remove 14480 multiallelic variants
2024-10-15 12:33:41 INFO::Harmonize 9127598 variants between GWAS and the reference
2024-10-15 12:33:58 INFO::Flip signs for 1272047 (13.94%) variants
2024-10-15 12:33:59 INFO::Remove 1272047 strand ambiguous variants
2024-10-15 12:33:59 INFO::7855551 variants left after preprocessing and harmonization.</code></pre>
<pre class="r"><code>dim(z_snp)</code></pre>
<pre><code>[1] 7855551       4</code></pre>
</div>
<div id="prediction-model" class="section level3">
<h3>Prediction model</h3>
<p>The prediction model was accessed from GTEx. The molecular traits were expression and splicing. Here, I used artery aorta as example.</p>
<pre class="r"><code>artery_eqtl &lt;- &quot;mashr_eqtl/eqtl/mashr/mashr_Artery_Aorta.db&quot;
artery_eqtl_weights &lt;- preprocess_weights(artery_eqtl,
                                    region_info,
                                    z_snp$id,
                                    snp_map,
                                    type = &quot;expression&quot;,
                                    context = &quot;artery&quot;)

artery_sqtl &lt;- &quot;mashr_sqtl/sqtl/mashr/mashr_Artery_Aorta.db&quot;
artery_sqtl_weights &lt;- preprocess_weights(artery_sqtl,
                                          region_info,
                                          z_snp$id,
                                          snp_map,
                                          type = &quot;splicing&quot;,
                                          context = &quot;artery&quot;)

weights &lt;- c(artery_eqtl_weights, artery_sqtl_weights)</code></pre>
</div>
</div>
<div id="run-ctwas" class="section level2">
<h2>Run cTWAS</h2>
<p>After all the input files ready, run the cTWAS.</p>
<pre class="r"><code># # ctwas_res &lt;- ctwas_sumstats_noLD(z_snp,
#                                  artery,
#                                  region_info,
#                                  snp_map,
#                                  thin = 0.1, 
#                                  ncore = 6)

load(&quot;ctwas_res_noLD.RData&quot;)

z_gene &lt;- ctwas_res$z_gene
param &lt;- ctwas_res$param
finemap_res &lt;- ctwas_res$finemap_res
susie_alpha_res &lt;- ctwas_res$susie_alpha_res
boundary_genes &lt;- ctwas_res$boundary_genes
region_data &lt;- ctwas_res$region_data
screen_res &lt;- ctwas_res$screen_res</code></pre>
<div id="explore-and-interpret-the-results" class="section level3">
<h3>Explore and interpret the results</h3>
<p><strong>Create the convergence plots:</strong></p>
<pre class="r"><code>gwas_n &lt;- 1165690
make_convergence_plots(param, gwas_n)</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>These plots show the estimated prior inclusion probability, prior effect size variance, enrichment and proportion of variance explained (PVE) over the iterations of parameter estimation.</p>
<p><strong>Estimate the parameters:</strong></p>
<pre class="r"><code>ctwas_parameters &lt;- summarize_param(param, gwas_n)
ctwas_parameters</code></pre>
<pre><code>$group_size
artery|expression   artery|splicing               SNP 
            10403             29657           7851620 

$group_prior
artery|expression   artery|splicing               SNP 
      0.008485956       0.006215176       0.000206954 

$group_prior_var
artery|expression   artery|splicing               SNP 
         50.79135          11.08500          10.61658 

$enrichment
artery|expression   artery|splicing 
         41.00406          30.03167 

$group_pve
artery|expression   artery|splicing               SNP 
      0.003846503       0.001752803       0.014799087 

$total_pve
[1] 0.02039839

$prop_heritability
artery|expression   artery|splicing               SNP 
       0.18856891        0.08592849        0.72550260 </code></pre>
<p><strong>Create the pie chart to see the heritability:</strong></p>
<pre class="r"><code>data &lt;- data.frame(
  category = names(ctwas_parameters$prop_heritability),
  percentage = ctwas_parameters$prop_heritability
)

data$percentage_label &lt;- paste0(round(data$percentage * 100, 1), &quot;%&quot;)

ggplot(data, aes(x = &quot;&quot;, y = percentage, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) +
  coord_polar(&quot;y&quot;, start = 0) +
  theme_void() +  # Remove background and axes
  geom_text(aes(label = percentage_label), 
            position = position_stack(vjust = 0.5), size = 5) +
  scale_fill_manual(values = c(&quot;#FF9999&quot;, &quot;#66B2FF&quot;, &quot;#99FF99&quot;)) +  # Custom colors
  labs(fill = &quot;Category&quot;) +
  ggtitle(&quot;Percent of heritability&quot;)</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/5b1f986a50470ba73ea2c2f212f0def4e63c3255/docs/figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-8-1.png" target="_blank">5b1f986</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Create a diagnostic plot:</strong></p>
<pre class="r"><code>ggplot(data = finemap_res[finemap_res$type!=&quot;SNP&quot;,], aes(x = abs(z), y = susie_pip)) +
  geom_point() +
  labs(x = &quot;abs(z-scores)&quot;, y = &quot;PIPs&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inspecting-and-summarizing-results" class="section level3">
<h3>Inspecting and summarizing results</h3>
<p><strong>Add p-values to finemapping results:</strong></p>
<pre class="r"><code>finemap_res$molecular_id &lt;- get_molecular_ids(finemap_res)
finemap_res$pval &lt;- z2p(finemap_res$z)
head(finemap_res)</code></pre>
<pre><code>                                    id       molecular_id       type context
1 ENSG00000163875.15|artery_expression ENSG00000163875.15 expression  artery
2 ENSG00000163877.10|artery_expression ENSG00000163877.10 expression  artery
3 ENSG00000163879.10|artery_expression ENSG00000163879.10 expression  artery
4 ENSG00000134697.12|artery_expression ENSG00000134697.12 expression  artery
5 ENSG00000116922.14|artery_expression ENSG00000116922.14 expression  artery
6 ENSG00000134690.10|artery_expression ENSG00000134690.10 expression  artery
              group           region_id          z    susie_pip       mu2   cs
1 artery|expression 1_37083582_38266175 -1.4716158 7.290964e-10  3.063522 &lt;NA&gt;
2 artery|expression 1_37083582_38266175 -0.3554597 2.682295e-10  1.102211 &lt;NA&gt;
3 artery|expression 1_37083582_38266175 -4.3682669 2.918403e-06 19.332691 &lt;NA&gt;
4 artery|expression 1_37083582_38266175  1.6427321 9.468295e-10  3.576057 &lt;NA&gt;
5 artery|expression 1_37083582_38266175  0.8119708 3.483370e-10  1.614774 &lt;NA&gt;
6 artery|expression 1_37083582_38266175  0.1591359 2.552656e-10  1.005047 &lt;NA&gt;
          pval
1 1.411247e-01
2 7.222452e-01
3 1.252364e-05
4 1.004384e-01
5 4.168084e-01
6 8.735618e-01</code></pre>
<p><strong>View the results: mol traits, PIP &gt; 0.8</strong></p>
<pre class="r"><code>finemap_sig &lt;- subset(finemap_res, group != &quot;SNP&quot; &amp; susie_pip &gt; 0.8 &amp; !is.na(cs))</code></pre>
<p>There were 18 significant molecular traits: 12 expression, 6 splicing.</p>
<p><strong>Compute gene PIPs:</strong></p>
<pre class="r"><code>combined_pip_by_context &lt;- combine_gene_pips(susie_alpha_res, 
                                             group_by = &quot;molecular_id&quot;,
                                             by = &quot;context&quot;,
                                             method = &quot;combine_cs&quot;,
                                             filter_cs = TRUE,
                                             include_cs_id = TRUE)

combined_pip_by_context_sig &lt;- subset(combined_pip_by_context, combined_pip &gt; 0.8)

DT::datatable(combined_pip_by_context_sig,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Significant genes&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-232a8e63d6aacd296363" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-232a8e63d6aacd296363">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Significant genes<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"],["ENSG00000273611.4","ENSG00000112137.17","intron_3_48694240_48695090","ENSG00000118526.6","ENSG00000182511.11","ENSG00000112655.15","ENSG00000158186.12","intron_19_43649143_43652225","ENSG00000136715.17","intron_7_45038426_45063918","ENSG00000087237.10","ENSG00000142039.3","ENSG00000107798.17","ENSG00000133433.10","ENSG00000119718.10","intron_4_94576034_94584986","intron_2_162267569_162268087","intron_5_132419940_132449650"],["17_36141651_38653091.L1","6_11790583_13209156.L1","3_47685722_49279539.L1","6_132444530_133923105.L1","15_89932319_91621162.L1","6_42070983_43788432.L1","3_137652241_140235755.L1","19_43358303_44239955.L1","2_127276771_128716867.L1","7_44724229_45913323.L1","16_55869862_57630418.L1","19_40478694_41625199.L1","10_87367307_89253624.L1","22_23370460_24588236.L1","14_72422907_75978424.L1","4_93312148_95261037.L1","2_160913222_162647041.L1","5_130183332_132803957.L1"],[0.9999999987967522,0.9999999982747347,0.9985774149213016,0.9947064457627866,0.9880651861874525,0.9828981429256251,0.9681278599695083,0.9656954876628852,0.9622142043943018,0.9618786208606445,0.9506820580616597,0.9365605088437765,0.9292908036104005,0.9142060080052019,0.8518815579766307,0.8506441728893748,0.8464776700275961,0.8257350502215761],["17_36141651_38653091.L1","6_11790583_13209156.L1","3_47685722_49279539.L1","6_132444530_133923105.L1","15_89932319_91621162.L1","6_42070983_43788432.L1","3_137652241_140235755.L1","19_43358303_44239955.L1","2_127276771_128716867.L1","7_44724229_45913323.L1","16_55869862_57630418.L1","19_40478694_41625199.L1","10_87367307_89253624.L1","22_23370460_24588236.L1","14_72422907_75978424.L1","4_93312148_95261037.L1","2_160913222_162647041.L1","5_130183332_132803957.L1"],[0.9999999987967522,0.9999999982747347,0.9985774149213016,0.9947064457627866,0.9880651861874525,0.9828981429256251,0.9681278599695083,0.9656954876628852,0.9622142043943018,0.9618786208606445,0.9506820580616597,0.9365605088437765,0.9292908036104005,0.9142060080052019,0.8518815579766307,0.8506441728893748,0.8464776700275961,0.8257350502215761]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>molecular_id<\/th>\n      <th>combined_cs_id<\/th>\n      <th>combined_pip<\/th>\n      <th>artery_cs_id<\/th>\n      <th>artery_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><strong>Add gene annotations:(make sure the genome build match)</strong></p>
<pre class="r"><code>ens_db &lt;- EnsDb.Hsapiens.v86
mapping_table &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS&quot;)</code></pre>
<p><strong>Update finemapping results with gene annotation:</strong></p>
<pre class="r"><code># Update finemap results with gene annot
finemap_res &lt;- anno_finemap_res(finemap_res,
                                snp_map = snp_map,
                                mapping_table = mapping_table,
                                add_gene_annot = TRUE,
                                map_by = &quot;molecular_id&quot;,
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-15 12:34:16 INFO::Annotating fine-mapping result ...
2024-10-15 12:34:16 INFO::Map molecular traits to genes
2024-10-15 12:34:17 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-15 12:34:18 INFO::Add gene positions
2024-10-15 12:34:18 INFO::Add SNP positions</code></pre>
<pre class="r"><code>finemap_res_sig &lt;- subset(finemap_res, group != &quot;SNP&quot; &amp; gene_type == &quot;protein_coding&quot; &amp; susie_pip &gt; 0.8 &amp; !is.na(cs))

DT::datatable(finemap_res_sig,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Significant protein-coding genes&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-2112c1f1dbc2c6427f4c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2112c1f1dbc2c6427f4c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Significant protein-coding genes<\/caption>","data":[["978","1061","1241","1756","1946","2255","2402","2538","2664","3392","4898","5418","5701","6631","7599","8715"],["ENSG00000136715.17|artery_expression","intron_2_162267569_162268087|artery_splicing","intron_3_48694240_48695090|artery_splicing","ENSG00000158186.12|artery_expression","intron_4_94576034_94584986|artery_splicing","ENSG00000112137.17|artery_expression","ENSG00000112655.15|artery_expression","ENSG00000118526.6|artery_expression","intron_7_45038426_45063918|artery_splicing","ENSG00000107798.17|artery_expression","ENSG00000119718.10|artery_expression","ENSG00000182511.11|artery_expression","ENSG00000087237.10|artery_expression","ENSG00000273611.4|artery_expression","ENSG00000142039.3|artery_expression","ENSG00000133433.10|artery_expression"],["ENSG00000136715.17","intron_2_162267569_162268087","intron_3_48694240_48695090","ENSG00000158186.12","intron_4_94576034_94584986","ENSG00000112137.17","ENSG00000112655.15","ENSG00000118526.6","intron_7_45038426_45063918","ENSG00000107798.17","ENSG00000119718.10","ENSG00000182511.11","ENSG00000087237.10","ENSG00000273611.4","ENSG00000142039.3","ENSG00000133433.10"],["expression","splicing","splicing","expression","splicing","expression","expression","expression","splicing","expression","expression","expression","expression","expression","expression","expression"],["artery","artery","artery","artery","artery","artery","artery","artery","artery","artery","artery","artery","artery","artery","artery","artery"],["artery|expression","artery|splicing","artery|splicing","artery|expression","artery|splicing","artery|expression","artery|expression","artery|expression","artery|splicing","artery|expression","artery|expression","artery|expression","artery|expression","artery|expression","artery|expression","artery|expression"],["2_127276771_128716867","2_160913222_162647041","3_47685722_49279539","3_137652241_140235755","4_93312148_95261037","6_11790583_13209156","6_42070983_43788432","6_132444530_133923105","7_44724229_45913323","10_87367307_89253624","14_72422907_75978424","15_89932319_91621162","16_55869862_57630418","17_36141651_38653091","19_40478694_41625199","22_23370460_24588236"],[5.567700598040783,4.944459162508342,-7.251970456285184,-8.778096024375058,5.959842485928168,-22.7047586153854,-6.180534334073398,-11.28712877513667,-5.658592810454961,-10.90450535565776,6.95024067499966,11.90468531223122,-6.772756041508129,7.666918762224385,9.051407355881695,-6.269548147605076],[0.9622142043943018,0.8464776700275961,0.9985774149213016,0.9681278599695083,0.8506441728893748,0.9999999982747347,0.9828981429256251,0.9947064457627866,0.9618786208606445,0.9292908036104005,0.8518815579766307,0.9880651861874525,0.9506820580616597,0.9999999987967522,0.9365605088437765,0.9142060080052019],[30.7944548202521,21.48636969833871,45.16489264192939,75.08879605959332,30.80186480021843,496.771907252915,37.71882594935626,123.5077507618006,27.85708863045054,115.3414403919602,47.43914396313349,137.2822713862154,45.09667002916034,57.514308781233,79.7754397190696,38.78467250374324],["L1","L1","L1","L1","L1","L1","L1","L1","L1","L1","L1","L1","L1","L1","L1","L1"],[2.581228438112767e-08,7.635554492348005e-07,4.107514957326666e-13,1.662652924361874e-18,2.524811795024659e-09,4.020363419219014e-114,6.388499997692151e-10,1.51918553805017e-29,1.526191757500281e-08,1.096882027994512e-27,3.646635491062325e-12,1.118845822262273e-32,1.263519150773118e-11,1.761772830056233e-14,1.411373788902543e-19,3.620973197954315e-10],["SAP130","IFIH1","IP6K2","MRAS","PDLIM5","PHACTR1","PTK7","TCF21","CCM2","LIPA","EIF2B2","FES","CETP","ZNHIT3","CCDC97","GSTT2B"],["protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding"],["2","2","3","3","4","6","6","6","7","10","14","15","16","17","19","22"],["127941217","162267569","48694240","138347648","94576034","12716805","43076268","133889431","45038426","89213569","75002911","90883695","56961850","36486629","41310189","23957414"],["128028120","162268087","48695090","138405534","94584986","13290017","43161719","133895553","45063918","89414557","75011268","90895776","56983845","36499310","41324883","23961186"],[127984668,162267828,48694665,138376591,94580510,13003411,43118994,133892492,45051172,89314063,75007090,90889736,56972848,36492970,41317536,23959300]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>molecular_id<\/th>\n      <th>type<\/th>\n      <th>context<\/th>\n      <th>group<\/th>\n      <th>region_id<\/th>\n      <th>z<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>cs<\/th>\n      <th>pval<\/th>\n      <th>gene_name<\/th>\n      <th>gene_type<\/th>\n      <th>chrom<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>pos<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[7,8,9,11,17]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Compute gene PIPs across different types of molecular traits. Here, “group_by = gene_name”, because molecular traits are mapped to their corresponding genes by the “gene_name” column.</p>
<pre class="r"><code>susie_alpha_res &lt;- anno_susie_alpha_res(susie_alpha_res,
                                        mapping_table = mapping_table,
                                        map_by = &quot;molecular_id&quot;,
                                        drop_unmapped = TRUE)</code></pre>
<pre><code>2024-10-15 12:34:24 INFO::Annotating susie alpha result ...
2024-10-15 12:34:24 INFO::Map molecular traits to genes
2024-10-15 12:34:24 INFO::Split PIPs for molecular traits mapped to multiple genes</code></pre>
<pre class="r"><code>combined_pip_by_type &lt;- combine_gene_pips(susie_alpha_res,
                                          group_by = &quot;gene_name&quot;,
                                          by = &quot;type&quot;,
                                          method = &quot;combine_cs&quot;,
                                          filter_cs = TRUE,
                                          include_cs_id = TRUE)

combined_pip_by_type_sig &lt;- subset(combined_pip_by_type, combined_pip &gt; 0.8)

DT::datatable(combined_pip_by_type_sig,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Significant genes&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-57a5e797e447beab7be8" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-57a5e797e447beab7be8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Significant genes<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"],["ZNHIT3","PHACTR1","IP6K2","TCF21","ITGA1","ITIH4","FES","LIPA","PTK7","PDLIM5","MRAS","SAP130","CCM2","CETP","CCDC97","ARVCF","GSTT2B","ZSCAN31","CORO6","BMP1","EIF2B2","IFIH1"],["17_36141651_38653091.L1","6_11790583_13209156.L1","3_47685722_49279539.L1","6_132444530_133923105.L1","5_52838788_54407214.L1","3_51797999_54047363.L1","15_89932319_91621162.L1","10_87367307_89253624.L1","6_42070983_43788432.L1","4_93312148_95261037.L1","3_137652241_140235755.L1","2_127276771_128716867.L1","7_44724229_45913323.L1","16_55869862_57630418.L1","19_40478694_41625199.L1","22_19924835_22002927.L1","22_23370460_24588236.L1","6_28050041_28949831.L1","17_29007226_31459473.L1","8_21804225_23039544.L1","14_72422907_75978424.L1","2_160913222_162647041.L1"],[0.9999999987967522,0.9999999982747347,0.9985774149213016,0.9947064457627866,0.9909157127994216,0.9892821508808181,0.9880651861874525,0.9854237384006895,0.9828981429256251,0.9790080439141569,0.9681278599695083,0.9622142043943018,0.9618786208606445,0.9506820580616597,0.9365605088437765,0.9212489216996851,0.9142060080052019,0.8836261784322907,0.8718927849029132,0.862945568026024,0.8518815579766307,0.8464776700275961],["17_36141651_38653091.L1","6_11790583_13209156.L1",null,"6_132444530_133923105.L1",null,null,"15_89932319_91621162.L1","10_87367307_89253624.L1","6_42070983_43788432.L1",null,"3_137652241_140235755.L1","2_127276771_128716867.L1",null,"16_55869862_57630418.L1","19_40478694_41625199.L1","22_19924835_22002927.L1","22_23370460_24588236.L1",null,"17_29007226_31459473.L1","8_21804225_23039544.L1","14_72422907_75978424.L1",null],[0.9999999987967522,0.9999999982747347,null,0.9947064457627866,null,null,0.9880651861874525,0.9292908036104005,0.9828981429256251,null,0.9681278599695083,0.9622142043943018,null,0.9506820580616597,0.9365605088437765,0.5240436172219155,0.9142060080052019,null,0.5854079452182404,0.284631826202467,0.8518815579766307,null],[null,null,"3_47685722_49279539.L1",null,"5_52838788_54407214.L1","3_51797999_54047363.L1",null,"10_87367307_89253624.L1",null,"4_93312148_95261037.L1",null,null,"7_44724229_45913323.L1",null,null,"22_19924835_22002927.L1",null,"6_28050041_28949831.L1","17_29007226_31459473.L1","8_21804225_23039544.L1",null,"2_160913222_162647041.L1"],[null,null,0.9985774149213016,null,0.9909157127994216,0.9892821508808181,null,0.05613293479028902,null,0.9790080439141569,null,null,0.9618786208606445,null,null,0.3972053044777696,null,0.8836261784322907,0.2864848396846729,0.578313741823557,null,0.8464776700275961]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_cs_id<\/th>\n      <th>combined_pip<\/th>\n      <th>expression_cs_id<\/th>\n      <th>expression_pip<\/th>\n      <th>splicing_cs_id<\/th>\n      <th>splicing_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualize-by-the-gene-locus-plots" class="section level3">
<h3>Visualize by the gene locus plots</h3>
<pre class="r"><code>load(&quot;weights.RData&quot;)
make_locusplot(finemap_res,
               region_id = &quot;17_36141651_38653091&quot;,
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;)</code></pre>
<pre><code>2024-10-15 12:34:25 INFO::Limit to protein coding genes
2024-10-15 12:34:25 INFO::focal id: ENSG00000273611.4|artery_expression
2024-10-15 12:34:25 INFO::focal molecular trait: ZNHIT3 artery expression
2024-10-15 12:34:25 INFO::Range of locus: chr17:36143579-38652996</code></pre>
<pre><code>chromosome 17, position 36143579 to 38652996</code></pre>
<pre><code>4311 SNPs/datapoints</code></pre>
<pre><code>2024-10-15 12:34:25 INFO::focal molecular trait QTL positions: 36486677,36488913
2024-10-15 12:34:25 INFO::Limit PIPs to credible sets</code></pre>
<pre><code>Warning: ggrepel: 48 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/CAD_cTWAS_noLD.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The top track showed -log10(p-value) of the association of variants (from CAD GWAS) and molecular traits (from the package computed z-scores) with the phenotype.</p>
<p>The mid track showed PIPs of variants and molecular traits. We limited PIP results to credible sets in the PIP track (filter_cs = TRUE).</p>
<p>The next track showed the QTLs of the focal gene. We chose the molecular trait with the highest PIP (“PHACTR1” in this case).</p>
<p>The bottom is the gene track. We limited results to protein coding genes by default. We could draw a red dotted line to show the PIP cutoff of 0.8 by setting highlight_pip = 0.8.</p>
<p>As conclusion, we found PHACTR1 as the likely causal gene of CAD. The protein encoded by this gene is a member of the phosphatase and actin regulator family of proteins. Polymorphisms in this gene are associated with susceptibility to myocardial infarction, coronary artery disease and cervical artery dissection. Alternative splicing of this gene results in multiple transcript variants. The detailed gene function and summary of this gene can be found at <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=PHACTR1">genecard</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux 8.4 (Ootpa)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el8-x86_64/lib/libopenblas_skylakexp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.4.3             EnsDb.Hsapiens.v86_2.99.0
 [3] ensembldb_2.22.0          AnnotationFilter_1.22.0  
 [5] GenomicFeatures_1.50.4    AnnotationDbi_1.60.2     
 [7] Biobase_2.58.0            GenomicRanges_1.50.2     
 [9] GenomeInfoDb_1.34.9       IRanges_2.32.0           
[11] S4Vectors_0.36.2          BiocGenerics_0.44.0      
[13] RSQLite_2.3.1             ctwas_0.4.15             
[15] data.table_1.14.8         workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.38.0              locuszoomr_0.3.5           
  [7] fs_1.5.2                    rstudioapi_0.14            
  [9] farver_2.1.1                DT_0.22                    
 [11] ggrepel_0.9.3               bit64_4.0.5                
 [13] fansi_1.0.4                 xml2_1.3.3                 
 [15] codetools_0.2-18            logging_0.10-108           
 [17] cachem_1.0.8                knitr_1.42                 
 [19] jsonlite_1.8.7              Rsamtools_2.14.0           
 [21] dbplyr_2.3.2                png_0.1-7                  
 [23] readr_2.1.4                 compiler_4.2.0             
 [25] httr_1.4.7                  Matrix_1.6-4               
 [27] fastmap_1.1.1               lazyeval_0.2.2             
 [29] cli_3.6.2                   later_1.3.0                
 [31] htmltools_0.5.7             prettyunits_1.1.1          
 [33] tools_4.2.0                 gtable_0.3.4               
 [35] glue_1.6.2                  GenomeInfoDbData_1.2.9     
 [37] dplyr_1.1.3                 rappdirs_0.3.3             
 [39] Rcpp_1.0.11                 jquerylib_0.1.4            
 [41] vctrs_0.6.3                 Biostrings_2.66.0          
 [43] rtracklayer_1.58.0          crosstalk_1.2.0            
 [45] xfun_0.40                   stringr_1.5.0              
 [47] ps_1.7.0                    lifecycle_1.0.4            
 [49] irlba_2.3.5                 restfulr_0.0.15            
 [51] XML_3.99-0.9                getPass_0.2-2              
 [53] zlibbioc_1.44.0             zoo_1.8-11                 
 [55] scales_1.2.1                gggrid_0.2-0               
 [57] hms_1.1.3                   promises_1.2.0.1           
 [59] MatrixGenerics_1.10.0       ProtGenerics_1.30.0        
 [61] parallel_4.2.0              SummarizedExperiment_1.28.0
 [63] LDlinkR_1.3.0               yaml_2.3.7                 
 [65] curl_4.3.2                  memoise_2.0.1              
 [67] sass_0.4.1                  biomaRt_2.54.1             
 [69] stringi_1.7.12              highr_0.9                  
 [71] BiocIO_1.8.0                filelock_1.0.2             
 [73] BiocParallel_1.32.6         rlang_1.1.1                
 [75] pkgconfig_2.0.3             matrixStats_1.2.0          
 [77] bitops_1.0-7                evaluate_0.15              
 [79] lattice_0.20-45             purrr_1.0.2                
 [81] labeling_0.4.3              GenomicAlignments_1.34.1   
 [83] htmlwidgets_1.6.2           cowplot_1.1.1              
 [85] bit_4.0.5                   processx_3.5.3             
 [87] tidyselect_1.2.0            magrittr_2.0.3             
 [89] R6_2.5.1                    generics_0.1.3             
 [91] DelayedArray_0.24.0         DBI_1.1.3                  
 [93] withr_2.5.0                 pgenlibr_0.3.6             
 [95] pillar_1.9.0                whisker_0.4                
 [97] KEGGREST_1.38.0             RCurl_1.98-1.12            
 [99] mixsqp_0.3-54               tibble_3.2.1               
[101] crayon_1.5.2                utf8_1.2.3                 
[103] BiocFileCache_2.6.1         plotly_4.10.0              
[105] tzdb_0.4.0                  rmarkdown_2.21             
[107] progress_1.2.2              grid_4.2.0                 
[109] blob_1.2.4                  callr_3.7.0                
[111] git2r_0.30.1                digest_0.6.33              
[113] tidyr_1.3.0                 httpuv_1.6.5               
[115] munsell_0.5.0               viridisLite_0.4.2          
[117] bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
