<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xizhi Xu" />

<meta name="date" content="2024-10-25" />

<title>Check the novel discoveries</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CAD_cTWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xizhixu1016/CAD_cTWAS">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Check the novel discoveries</h1>
<h4 class="author">Xizhi Xu</h4>
<h4 class="date">10/25/2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CAD_cTWAS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241004code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20241004)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241004code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241004)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxizhixu1016CADcTWAStree3fe35939dbcfc1468cf5f233dcea463487f7d66dtargetblank3fe3593a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/xizhixu1016/CAD_cTWAS/tree/3fe35939dbcfc1468cf5f233dcea463487f7d66d" target="_blank">3fe3593</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomxizhixu1016CADcTWAStree3fe35939dbcfc1468cf5f233dcea463487f7d66dtargetblank3fe3593a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/xizhixu1016/CAD_cTWAS/tree/3fe35939dbcfc1468cf5f233dcea463487f7d66d" target="_blank">3fe3593</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  all_artery_weights.RData
    Untracked:  combined_pip_by_type_sig.RData
    Untracked:  ctwas_res_all.RData
    Untracked:  ctwas_res_noLD.RData
    Untracked:  res.RData
    Untracked:  weights.RData
    Untracked:  weights_group.RData
    Untracked:  z_snp.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/check_novel_genes.Rmd</code>) and HTML (<code>docs/check_novel_genes.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/3fe35939dbcfc1468cf5f233dcea463487f7d66d/docs/check_novel_genes.html" target="_blank">3fe3593</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xizhixu1016/CAD_cTWAS/fbbaca10280aff266f7a1366971bfc4d86028a99/docs/check_novel_genes.html" target="_blank">fbbaca1</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xizhixu1016/CAD_cTWAS/blob/77597f20f12a9e16f0993f18cf82bd5ab5b3d969/analysis/check_novel_genes.Rmd" target="_blank">77597f2</a>
</td>
<td>
Xizhi
</td>
<td>
2024-10-28
</td>
<td>
Create a new page to check whether novel genes are discovered
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>First, load the required packages and previous results.</p>
<pre class="r"><code>library(ctwas)
library(data.table)

load(&quot;combined_pip_by_type_sig.RData&quot;)

DT::datatable(combined_pip_by_type_sig,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Gene PIPs, only genes with credible sets are shown&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-232a8e63d6aacd296363" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-232a8e63d6aacd296363">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Gene PIPs, only genes with credible sets are shown<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36"],["TRMT61B","TBC1D7","ZNHIT3","PHACTR1","BOP1","NBEAL1","TCF21","IP6K2","ITIH4","PTK7","ITGA1","TBXAS1","SAP130","CCM2","MRAS","LIPA","TGFB1","PDLIM5","FES","CETP","CDK8","MAP1S","BMP1","CORO6","ARVCF","ZSCAN31","GSTT2B","SERPINH1","MIA3","SIPA1","R3HDM2","EIF2B2","RRBP1","C6orf106","PSMA3-AS1","CALCRL"],["2_28375910_28994693.L1","6_13209156_14802693.L1","17_36141651_38653091.L1","6_11790583_13209156.L1","8_143155464_145078481.L1","2_201953914_204934518.L1","6_132444530_133923105.L1","3_47685722_49279539.L1","3_51797999_54047363.L1","6_42070983_43788432.L1","5_52838788_54407214.L1","7_139059933_140233377.L1","2_127276771_128716867.L1","7_44724229_45913323.L1","3_137652241_140235755.L1","10_87367307_89253624.L1","19_40478694_41625199.L1","4_93312148_95261037.L1","15_89932319_91621162.L1","16_55869862_57630418.L1","13_25210224_26710225.L1","19_16263605_18299052.L1","8_21804225_23039544.L1","17_29007226_31459473.L1","22_19924835_22002927.L1","6_28050041_28949831.L1","22_23370460_24588236.L1","11_74701306_77086163.L1","1_222056878_224750818.L1","11_64037097_66131160.L1","12_57155077_58914885.L1","14_72422907_75978424.L1","20_17506587_18991559.L1","6_33268720_35487979.L1","14_57014544_58981618.L1","2_184415446_189017339.L1"],[1,0.99999999999859,0.9999999986804733,0.9999999986665887,0.9999977207557993,0.9977419225293662,0.9949814265971395,0.9929120590689428,0.9911879286097685,0.9840105982456073,0.9829025810527523,0.982881331447818,0.9793128705289912,0.9758817472749568,0.9729755965556406,0.9661834465598135,0.9603705130413827,0.9571234468497776,0.9567800000313702,0.9530128976621716,0.9260935181543878,0.9260536033551483,0.9214673199012092,0.9057661566689741,0.8993058209690633,0.895432078569548,0.8896169670592035,0.8896100605462344,0.8874382897006723,0.8867861046286577,0.881179888062415,0.8699079096379659,0.8591130455347936,0.8533544401385483,0.8038404689519556,0.8019895317733428],[null,"6_13209156_14802693.L1","17_36141651_38653091.L1","6_11790583_13209156.L1","8_143155464_145078481.L1","2_201953914_204934518.L1","6_132444530_133923105.L1",null,null,"6_42070983_43788432.L1",null,null,"2_127276771_128716867.L1",null,"3_137652241_140235755.L1","10_87367307_89253624.L1","19_40478694_41625199.L1",null,"15_89932319_91621162.L1","16_55869862_57630418.L1","13_25210224_26710225.L1","19_16263605_18299052.L1","8_21804225_23039544.L1","17_29007226_31459473.L1","22_19924835_22002927.L1",null,"22_23370460_24588236.L1","11_74701306_77086163.L1",null,null,null,"14_72422907_75978424.L1",null,"6_33268720_35487979.L1",null,"2_184415446_189017339.L1"],[null,0.99999999999859,0.9999999986804733,0.9999999986665887,0.9999977207557993,0.9977419225293662,0.9949814265971395,null,null,0.9840105982456073,null,null,0.9793128705289912,null,0.9729755965556406,0.929975873549565,0.9603705130413827,null,0.9567800000313702,0.9530128976621716,0.9260935181543878,0.7949466545016703,0.4335990366357124,0.4503065441917729,0.6548101610864523,null,0.8896169670592035,0.8896100605462344,null,null,null,0.8608294314926849,null,0.4284729044785814,null,0.7928952550351068],["2_28375910_28994693.L1",null,null,null,null,null,null,"3_47685722_49279539.L1","3_51797999_54047363.L1",null,"5_52838788_54407214.L1","7_139059933_140233377.L1",null,"7_44724229_45913323.L1",null,"10_87367307_89253624.L1",null,"4_93312148_95261037.L1",null,null,null,"19_16263605_18299052.L1","8_21804225_23039544.L1","17_29007226_31459473.L1","22_19924835_22002927.L1","6_28050041_28949831.L1",null,null,"1_222056878_224750818.L1","11_64037097_66131160.L1","12_57155077_58914885.L1","14_72422907_75978424.L1","20_17506587_18991559.L1","6_33268720_35487979.L1","14_57014544_58981618.L1","2_184415446_189017339.L1"],[1,null,null,null,null,null,null,0.9929120590689428,0.9911879286097685,null,0.9829025810527523,0.982881331447818,null,0.9758817472749568,null,0.03620757301024846,null,0.9571234468497776,null,null,null,0.131106948853478,0.4878682832654968,0.4554596124772012,0.2444956598826109,0.895432078569548,null,null,0.8874382897006723,0.8867861046286577,0.881179888062415,0.009078478145281066,0.8591130455347936,0.4248815356599669,0.8038404689519556,0.009094276738235929]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_cs_id<\/th>\n      <th>combined_pip<\/th>\n      <th>expression_cs_id<\/th>\n      <th>expression_pip<\/th>\n      <th>splicing_cs_id<\/th>\n      <th>splicing_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Now, let’s find if any of found genes are novel discoveries.</p>
<div id="trmt61b" class="section level3">
<h3>TRMT61B</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> TRMT61B encodes an enzyme involved in mitochondrial RNA modification, specifically in the methylation of mitochondrial tRNAs. The connection between mitochondrial health and cardiovascular conditions, however, suggests that TRMT61B could be of indirect interest, particularly if mitochondrial dysfunction emerges as a significant contributing factor to CAD in specific contexts.</p>
</div>
<div id="tbc1d7" class="section level3">
<h3>TBC1D7</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> The gene TBC1D7 has been implicated in coronary artery disease. Some studies have identified variants in TBC1D7 associated with cardiovascular traits.</p>
</div>
<div id="znhit3" class="section level3">
<h3>ZNHIT3</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> Some genetic association studies have identified ZNHIT3 as a candidate gene linked to CAD and other cardiovascular traits, particularly due to its potential roles in various biological processes relevant to cardiovascular health.</p>
</div>
<div id="phactr1" class="section level3">
<h3>PHACTR1</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> Numerous genome-wide association studies (GWAS) have identified PHACTR1 as a significant locus associated with CAD and related cardiovascular traits. Variants near or within the PHACTR1 gene have been consistently linked to increased risk of CAD and other cardiovascular conditions. PHACTR1 is involved in regulating the actin cytoskeleton and cellular signaling pathways, influencing cell migration, adhesion, and proliferation. These processes are essential for vascular remodeling and the development of atherosclerosis.</p>
</div>
<div id="bop1" class="section level3">
<h3>BOP1</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> GOP1 is a protein involved in cellular processes linked to the Golgi apparatus, which is a critical organelle in the modification, sorting, and transport of proteins and lipids. Proper Golgi function is crucial for endothelial cells, which line the blood vessels. Dysfunction in proteins involved in the Golgi apparatus could affect endothelial integrity, permeability, and the ability to regulate blood pressure and inflammation, all of which are relevant to CAD.</p>
</div>
<div id="nbeal1" class="section level3">
<h3>NBEAL1</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> NBEAL1 belongs to the BEACH (Beige and Chediak-Higashi) domain-containing protein family, which is involved in intracellular membrane trafficking, vesicle formation, and secretion. Several genome-wide association studies (GWAS) have identified NBEAL1 as a genetic locus associated with CAD risk. Variants in the NBEAL1 gene region have been linked to increased susceptibility to atherosclerosis and cardiovascular disease in diverse populations.</p>
</div>
<div id="tcf21" class="section level3">
<h3>TCF21</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> TCF21 (Transcription Factor 21) is a well-established gene involved in coronary artery disease (CAD). Research has identified TCF21 as a key player in cardiovascular development, and its role in CAD is linked to vascular biology and the formation of atherosclerotic plaques.</p>
</div>
<div id="ip6k2" class="section level3">
<h3>IP6K2</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> IP6K2 is an enzyme that converts inositol hexakisphosphate (IP6) to inositol pyrophosphates, specifically IP7 (5-diphosphoinositol pentakisphosphate). These molecules act as important second messengers in a variety of cellular processes. IP6K2’s role in apoptosis suggests it could influence plaque stability indirectly, impacting the risk of acute coronary events like heart attacks.</p>
</div>
<div id="itih4" class="section level3">
<h3>ITIH4</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> ITIH4 (Inter-Alpha-Trypsin Inhibitor Heavy Chain 4) has been studied in relation to various health conditions, including cardiovascular diseases like coronary artery disease (CAD). It is a protein primarily involved in inflammation, extracellular matrix stability, and the acute phase response, making it a potentially relevant player in the development of CAD.</p>
</div>
<div id="ptk7" class="section level3">
<h3>PTK7</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> PTK7 is involved in the Wnt signaling pathway, which is crucial for cell development, differentiation, and migration. Inflammation plays a significant role in CAD, contributing to the initiation and progression of atherosclerosis. PTK7’s regulation of cell signaling pathways, such as the Wnt pathway, can affect inflammatory responses in the cardiovascular system.</p>
</div>
<div id="itga1" class="section level3">
<h3>ITGA1</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> ITGA1 (Integrin Subunit Alpha 1) is a gene that encodes a protein that is part of the integrin family. ITGA1, through its role in ECM interactions, can influence how cells respond to the ECM during plaque development. Dysregulation of integrin signaling could affect plaque stability, leading to an increased risk of rupture and acute coronary events.</p>
</div>
<div id="tbxas1" class="section level3">
<h3>TBXAS1</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> Genetic variants in the TBXAS1 gene have been investigated in relation to cardiovascular risk. Some studies have found associations between certain variants and increased susceptibility to thrombotic events, which are relevant to CAD.</p>
</div>
<div id="sap130" class="section level3">
<h3>SAP130</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> SAP130 (Sin3A-Associated Protein, 130 kDa) is a gene that encodes a component of the Sin3A transcriptional corepressor complex. This complex is involved in chromatin remodeling and gene expression regulation, affecting various biological processes, including cell cycle control, apoptosis, and immune responses. SAP130, through its influence on immune activation and inflammation, may indirectly contribute to the endothelial dysfunction seen in early stages of CAD.</p>
</div>
<div id="ccm2" class="section level3">
<h3>CCM2</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> CCM2 (Cerebral Cavernous Malformation 2) is a gene that encodes a protein involved in the maintenance of vascular integrity and regulation of endothelial cell function. CCM2’s role in vascular integrity and endothelial function suggests it potentially relevant in CAD.</p>
</div>
<div id="mras" class="section level3">
<h3>MRAS</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> MRAS (M Ras) is a gene that encodes a member of the Ras family of GTPases, which are involved in various cellular signaling pathways. MRAS’s role in regulating endothelial cell adhesion and migration may be crucial in maintaining endothelial integrity, and disruptions in its function could contribute to endothelial dysfunction associated with CAD.</p>
</div>
<div id="lipa" class="section level3">
<h3>LIPA</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> LIPA (Lysosomal Acid Lipase) is a gene that encodes an enzyme responsible for the breakdown of lipids. Genetic variants in the LIPA gene have been associated with altered lipid profiles and increased risk for CAD. Studies have shown that certain polymorphisms can impact LIPA enzyme activity and lipid metabolism.</p>
</div>
<div id="tgfb1" class="section level3">
<h3>TGFB1</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> TGFB1 (Transforming Growth Factor Beta 1) is a cytokine that plays a crucial role in various biological processes. TGFB1 is implicated in the pathogenesis of atherosclerosis, a major contributor to CAD. Chronic exposure to high levels of TGFB1 can lead to endothelial dysfunction, characterized by impaired nitric oxide production and increased expression of adhesion molecules.</p>
</div>
<div id="pdlim5" class="section level3">
<h3>PDLIM5</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> PDLIM5 (PDZ and LIM Domain Containing 5) is a gene that encodes a protein belonging to the PDZ-LIM family. Given PDLIM5’s involvement in endothelial cell adhesion and signaling, alterations in its function could potentially contribute to endothelial dysfunction, an early event in CAD development. PDLIM5’s role in regulating cell adhesion and migration suggests it could influence the development of atherosclerosis.</p>
</div>
<div id="fes" class="section level3">
<h3>FES</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> FES (Feline Sarcoma Oncogene) encodes a non-receptor tyrosine kinase involved in signaling pathways that regulate cell growth, differentiation, and survival. Given FES’s role in immune cell regulation, it may contribute to inflammatory responses within atherosclerotic plaques. Dysregulated FES activity might impair endothelial repair, promoting an environment conducive to CAD progression.</p>
</div>
<div id="cetp" class="section level3">
<h3>CETP</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> CETP (Cholesteryl Ester Transfer Protein) is involved in coronary artery disease (CAD) and has been extensively studied for its role in lipid metabolism. CETP activity impacts cholesterol levels and distribution within the bloodstream, which can directly influence the development of atherosclerosis, a major contributor to CAD.</p>
</div>
<div id="cdk8" class="section level3">
<h3>CDK8</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> CDK8 (Cyclin-Dependent Kinase 8) is primarily known as a regulatory kinase involved in gene transcription, particularly in modifying RNA polymerase II activity and mediating transcriptional responses to various signaling pathways. CDK8’s regulation of inflammation-related genes suggests it may contribute to atherosclerosis, the buildup of plaques in arterial walls. Dysregulation of inflammation is a major factor in CAD, and genes modulating these processes can influence CAD susceptibility.</p>
</div>
<div id="map1s" class="section level3">
<h3>MAP1S</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> MAP1S (Microtubule-Associated Protein 1S) is a gene encoding a protein involved in cellular processes like autophagy, mitochondrial quality control, and microtubule stability. Impaired MAP1S function could lead to accumulated mitochondrial damage, increasing oxidative stress and promoting vascular injury and inflammation, both critical in CAD.</p>
</div>
<div id="bmp1" class="section level3">
<h3>BMP1</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> BMP1 (Bone Morphogenetic Protein 1) is involved in processes that are relevant to coronary artery disease (CAD) and has been implicated in cardiovascular health through its roles in extracellular matrix (ECM) remodeling and tissue repair.</p>
</div>
<div id="coro6" class="section level3">
<h3>CORO6</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> CORO6 (Coronin 6) is a member of the coronin family of proteins, which are known to be involved in various cellular processes, including actin cytoskeleton dynamics, cell signaling, and immune responses. Since chronic inflammation is a hallmark of atherosclerosis, CORO6’s role in regulating immune cell functions could affect inflammatory responses in the vascular wall. This could lead to changes in plaque stability and the progression of CAD.</p>
</div>
<div id="arvcf" class="section level3">
<h3>ARVCF</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> ARVCF (Armadillo Repeat Gene Deleted in Velocardiofacial Syndrome) is a member of the armadillo repeat family of proteins, which are involved in various cellular processes, including cell adhesion, signaling, and the regulation of the cytoskeleton. Given its role in cell adhesion and junction stability, ARVCF may influence endothelial cell function. Endothelial dysfunction is a key event in the pathogenesis of CAD, and proteins that regulate adhesion and barrier function are critical in maintaining vascular health.</p>
</div>
<div id="zscan31" class="section level3">
<h3>ZSCAN31</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> ZSCAN31 (Zinc Finger and SCAN Domain Containing 31) is a transcription factor that has been implicated in various biological processes, including gene regulation, stem cell maintenance, and development. ZSCAN31’s role in gene regulation and inflammatory responses suggests it could be involved in atherosclerotic processes. Chronic inflammation in the vascular wall is a hallmark of atherosclerosis, and proteins that modulate inflammatory pathways could influence CAD risk.</p>
</div>
<div id="gstt2b" class="section level3">
<h3>GSTT2B</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> GSTT2B (Glutathione S-Transferase Theta 2B) is part of the glutathione S-transferase (GST) family of enzymes, which are involved in detoxification processes by catalyzing the conjugation of glutathione to various substrates, including drugs and environmental toxins. Given GSTT2B’s role in detoxifying ROS, it may have protective effects against oxidative damage to endothelial cells. Endothelial dysfunction is a critical event in the development of CAD, and any protein that helps mitigate oxidative stress could potentially influence CAD risk.</p>
</div>
<div id="serpinh1" class="section level3">
<h3>SERPINH1</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> SERPINH1 (Serpin Family H Member 1), also known as heat shock protein 47 (HSP47), is a member of the serpin superfamily of proteins. Collagen is essential for maintaining endothelial integrity and function. Any alterations in collagen production or structure mediated by SERPINH1 could potentially impact endothelial health, which is critical in CAD development.</p>
</div>
<div id="mia3" class="section level3">
<h3>MIA3</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> MIA3 (Melanoma Inhibitory Activity 3), also known as TANGO1 (Transport and Golgi Organization Protein 1), is a protein involved in the trafficking and secretion of large extracellular matrix proteins, particularly collagens. MIA3’s role in collagen secretion may influence plaque stability in atherosclerosis. Collagen is essential for stabilizing atherosclerotic plaques; inadequate collagen in plaques can lead to rupture, which is a major cause of myocardial infarction in CAD.</p>
</div>
<div id="sipa1" class="section level3">
<h3>SIPA1</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> SIPA1 (Signal-Induced Proliferation-Associated 1) is a gene that encodes a protein involved in cell signaling and the regulation of cellular functions, including cell adhesion, migration, and proliferation. SIPA1’s regulation of cell adhesion and migration may influence endothelial cell function, which is vital for vascular health. Endothelial dysfunction is an early event in CAD, contributing to plaque formation and progression.</p>
</div>
<div id="r3hdm2" class="section level3">
<h3>R3HDM2</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> R3HDM2 (R3H Domain-Containing Protein 2) is a gene that encodes a protein with an R3H domain, a motif that enables RNA or single-stranded DNA binding. While R3HDM2 itself hasn’t been directly implicated in CAD, genes involved in regulating mRNA stability and cellular response to stress can influence cardiovascular health. For example, oxidative stress and inflammation are key drivers of CAD, and proteins that help cells manage these stressors may indirectly impact CAD risk.</p>
</div>
<div id="eif2b2" class="section level3">
<h3>EIF2B2</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> EIF2B2 (Eukaryotic Translation Initiation Factor 2B Subunit Beta) is a gene that encodes a subunit of the eIF2B complex, which plays a key role in the initiation of protein translation, particularly under cellular stress. By contributing to the cellular stress response, EIF2B2 could indirectly impact vascular cell resilience in CAD. Endothelial cells and vascular smooth muscle cells benefit from efficient stress adaptation, helping maintain vascular integrity under adverse conditions.</p>
</div>
<div id="rrbp1" class="section level3">
<h3>RRBP1</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> RRBP1 (Ribosome Binding Protein 1) encodes a protein associated with the rough endoplasmic reticulum (ER), primarily involved in the binding of ribosomes to the ER membrane and in protein translocation. In the context of CAD, vascular cells (such as endothelial cells and smooth muscle cells) require efficient ER function to cope with inflammatory and oxidative stress. By aiding in protein translocation and folding, RRBP1 might indirectly support the resilience of these cells, thereby contributing to plaque stability.</p>
</div>
<div id="c6orf106" class="section level3">
<h3>C6orf106</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> C6orf106 is thought to regulate immune signaling pathways, particularly those involved in inflammation. This role may involve influencing the stability of specific cytokine mRNAs, which are essential in mediating inflammatory responses.</p>
</div>
<div id="psma3-as1" class="section level3">
<h3>PSMA3-AS1</h3>
<p><strong>Novel:</strong> Yes</p>
<p><strong>Context:</strong> PSMA3-AS1 (PSMA3 Antisense RNA 1) is a long non-coding RNA (lncRNA) that is transcribed antisense to the PSMA3 gene, which encodes a subunit of the proteasome complex involved in protein degradation. Impaired proteostasis is associated with atherosclerosis, as the buildup of damaged proteins contributes to plaque formation and inflammation. If PSMA3-AS1 influences proteasome function, it could impact proteostasis in vascular cells, potentially affecting atherosclerosis progression and CAD risk.</p>
</div>
<div id="calcrl" class="section level3">
<h3>CALCRL</h3>
<p><strong>Novel:</strong> No</p>
<p><strong>Context:</strong> CALCRL (Calcitonin Receptor-Like Receptor) encodes a receptor that plays a key role in cardiovascular and inflammatory processes by mediating the effects of several peptides, including adrenomedullin and calcitonin gene-related peptide (CGRP). These peptides influence vascular tone, endothelial function, and inflammation, making CALCRL a gene of interest in studies on coronary artery disease (CAD).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux 8.4 (Ootpa)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el8-x86_64/lib/libopenblas_skylakexp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.14.8 ctwas_0.4.15      workflowr_1.7.1  

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.38.0              locuszoomr_0.3.5           
  [7] GenomicRanges_1.50.2        fs_1.5.2                   
  [9] rstudioapi_0.14             DT_0.22                    
 [11] ggrepel_0.9.3               bit64_4.0.5                
 [13] AnnotationDbi_1.60.2        fansi_1.0.4                
 [15] xml2_1.3.3                  codetools_0.2-18           
 [17] logging_0.10-108            cachem_1.0.8               
 [19] knitr_1.42                  jsonlite_1.8.7             
 [21] Rsamtools_2.14.0            dbplyr_2.3.2               
 [23] png_0.1-7                   readr_2.1.4                
 [25] compiler_4.2.0              httr_1.4.7                 
 [27] Matrix_1.6-4                fastmap_1.1.1              
 [29] lazyeval_0.2.2              cli_3.6.2                  
 [31] later_1.3.0                 htmltools_0.5.7            
 [33] prettyunits_1.1.1           tools_4.2.0                
 [35] gtable_0.3.4                glue_1.6.2                 
 [37] GenomeInfoDbData_1.2.9      dplyr_1.1.3                
 [39] rappdirs_0.3.3              Rcpp_1.0.11                
 [41] Biobase_2.58.0              jquerylib_0.1.4            
 [43] vctrs_0.6.3                 Biostrings_2.66.0          
 [45] rtracklayer_1.58.0          crosstalk_1.2.0            
 [47] xfun_0.40                   stringr_1.5.0              
 [49] ps_1.7.0                    lifecycle_1.0.4            
 [51] irlba_2.3.5                 restfulr_0.0.15            
 [53] ensembldb_2.22.0            XML_3.99-0.9               
 [55] getPass_0.2-2               zlibbioc_1.44.0            
 [57] zoo_1.8-11                  scales_1.2.1               
 [59] gggrid_0.2-0                hms_1.1.3                  
 [61] promises_1.2.0.1            MatrixGenerics_1.10.0      
 [63] ProtGenerics_1.30.0         parallel_4.2.0             
 [65] SummarizedExperiment_1.28.0 AnnotationFilter_1.22.0    
 [67] LDlinkR_1.3.0               yaml_2.3.7                 
 [69] curl_4.3.2                  memoise_2.0.1              
 [71] ggplot2_3.4.3               sass_0.4.1                 
 [73] biomaRt_2.54.1              stringi_1.7.12             
 [75] RSQLite_2.3.1               S4Vectors_0.36.2           
 [77] BiocIO_1.8.0                GenomicFeatures_1.50.4     
 [79] BiocGenerics_0.44.0         filelock_1.0.2             
 [81] BiocParallel_1.32.6         GenomeInfoDb_1.34.9        
 [83] rlang_1.1.1                 pkgconfig_2.0.3            
 [85] matrixStats_1.2.0           bitops_1.0-7               
 [87] evaluate_0.15               lattice_0.20-45            
 [89] purrr_1.0.2                 GenomicAlignments_1.34.1   
 [91] htmlwidgets_1.6.2           cowplot_1.1.1              
 [93] bit_4.0.5                   processx_3.5.3             
 [95] tidyselect_1.2.0            magrittr_2.0.3             
 [97] R6_2.5.1                    IRanges_2.32.0             
 [99] generics_0.1.3              DelayedArray_0.24.0        
[101] DBI_1.1.3                   pgenlibr_0.3.6             
[103] pillar_1.9.0                whisker_0.4                
[105] KEGGREST_1.38.0             RCurl_1.98-1.12            
[107] mixsqp_0.3-54               tibble_3.2.1               
[109] crayon_1.5.2                utf8_1.2.3                 
[111] BiocFileCache_2.6.1         plotly_4.10.0              
[113] tzdb_0.4.0                  rmarkdown_2.21             
[115] progress_1.2.2              grid_4.2.0                 
[117] blob_1.2.4                  callr_3.7.0                
[119] git2r_0.30.1                digest_0.6.33              
[121] tidyr_1.3.0                 httpuv_1.6.5               
[123] stats4_4.2.0                munsell_0.5.0              
[125] viridisLite_0.4.2           bslib_0.3.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
