---
title: "CAD cTWAS (no LD)"
author: "Xizhi Xu"
date: "10/09/2024"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

## Set up

Load the required packages. 

```{r setup, message=FALSE, warning=FALSE}
### Set up
setwd("/project/xinhe/xizhi/cTWAS_2024/CAD_cTWAS")
# setwd("xizhi_xin_midway3/cTWAS_2024/CAD_cTWAS")
# setwd("xizhi_xin_midway3/cTWAS_2024")
library(data.table)
library(ctwas)
library(RSQLite)
library(EnsDb.Hsapiens.v75)
library(ggplot2)
library(workflowr)
```


## Prepare the input: GWAS summary statistics, Reference (region info, SNP info), Prediction model (weights)

### GWAS Sum Stats 

The CAD summary statistics data I used was accessed from website: https://www.nature.com/articles/s41588-022-01233-6 . I filtered the original data to maintain the MAF between (0.01, 0.99).

```{r, eval=FALSE}
gwas_data <- fread("GCST90132314_buildGRCh37.tsv", sep = "\t")

# Filter to remain the variants with MAF (0.01, 0.99) 
gwas_filter <- gwas_data[gwas_data$effect_allele_frequency > 0.01, ]
gwas_filter <- gwas_filter[gwas_filter$effect_allele_frequency < 0.99, ]
gwas_filter$z <- gwas_filter$beta / gwas_filter$standard_error

gwas_map <- gwas_filter[, c("chromosome", "base_pair_location", "z")]

```


### Reference

Here I got the region info from the package cTWAS. SNP info was also accessed from reference panel of cTWAS website.
Another thing to notice: genome build needs to be matched. (Here: GRCh37 / hg19)

```{r, eval=FALSE}
# Region info
region_file <- system.file("extdata/ldetect", "EUR.b37.ldetect.regions.RDS", package = "ctwas")
region_info <- readRDS(region_file)

# SNP info
ref_snp_info <- fread("ukb_b37_0.1_var_info.Rvar", sep = "\t") # 9324048
class(ref_snp_info) <- "data.frame"

res <- create_snp_map(region_info, ref_snp_info, ncore = 6)
region_info <- res$region_info
snp_map <- res$snp_map

head(region_info)

ref_data <- read.table("ukb_b37_0.1_var_info.Rvar", header = TRUE, sep = "\t")

colnames(gwas_map)[colnames(gwas_map) == "base_pair_location"] <- "pos"
colnames(gwas_map)[colnames(gwas_map) == "chromosome"] <- "chrom"
head(gwas_map)

# Harmonize GWAS summary statistics.
z_snp <- merge(gwas_map, ref_data, by = c("pos", "chrom")) # 9155485
z_snp <- z_snp[, c("id", "alt", "ref", "z")]
colnames(z_snp)[colnames(z_snp) == "alt"] <- "A1"
colnames(z_snp)[colnames(z_snp) == "ref"] <- "A2"
head(z_snp)

z_snp <- preprocess_z_snp(z_snp, snp_map)

```

### Prediction model

The prediction model was accessed from GTEx. The molecular traits were expression and splicing. Here, I used artery aorta as example. 

```{r,eval=FALSE}
artery_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Aorta.db"
artery_eqtl_weights <- preprocess_weights(artery_eqtl,
                                    region_info,
                                    z_snp$id,
                                    snp_map,
                                    type = "expression",
                                    context = "artery")

artery_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Aorta.db"
artery_sqtl_weights <- preprocess_weights(artery_sqtl,
                                          region_info,
                                          z_snp$id,
                                          snp_map,
                                          type = "splicing",
                                          context = "artery")

weights <- c(artery_eqtl_weights, artery_sqtl_weights)
```


## Run cTWAS

After all the input files ready, run the cTWAS.

```{r}
# # ctwas_res <- ctwas_sumstats_noLD(z_snp,
#                                  artery,
#                                  region_info,
#                                  snp_map,
#                                  thin = 0.1, 
#                                  ncore = 6)

load("ctwas_res_noLD.RData")

ctwas_res <- ctwas_res2

z_gene <- ctwas_res$z_gene
param <- ctwas_res$param
finemap_res <- ctwas_res$finemap_res
susie_alpha_res <- ctwas_res$susie_alpha_res
boundary_genes <- ctwas_res$boundary_genes
region_data <- ctwas_res$region_data
screen_res <- ctwas_res$screen_res
```

### Explore and interpret the results

**Create the convergence plots:**

```{r}
gwas_n <- 1165690
make_convergence_plots(param, gwas_n)

```

These plots show the estimated prior inclusion probability, prior effect size variance, enrichment and proportion of variance explained (PVE) over the iterations of parameter estimation. 


**Estimate the parameters:**

```{r}
ctwas_parameters <- summarize_param(param, gwas_n)
ctwas_parameters
```


**Create the pie chart to see the heritability:**

```{r}
data <- data.frame(
  category = names(ctwas_parameters$prop_heritability),
  percentage = ctwas_parameters$prop_heritability
)

data$percentage_label <- paste0(round(data$percentage * 100, 1), "%")

ggplot(data, aes(x = "", y = percentage, fill = category)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +  # Remove background and axes
  geom_text(aes(label = percentage_label), 
            position = position_stack(vjust = 0.5), size = 5) +
  scale_fill_manual(values = c("#FF9999", "#66B2FF", "#99FF99")) +  # Custom colors
  labs(fill = "Category") +
  ggtitle("Percent of heritability")

```


**Create a diagnostic plot:**

```{r}
ggplot(data = finemap_res[finemap_res$type!="SNP",], aes(x = abs(z), y = susie_pip)) +
  geom_point() +
  labs(x = "abs(z-scores)", y = "PIPs") +
  theme_minimal()
```

### Inspecting and summarizing results

**Add p-values to finemapping results:**

```{r}
finemap_res$molecular_id <- get_molecular_ids(finemap_res)
finemap_res$pval <- z2p(finemap_res$z)
head(finemap_res)
```

**View the results: mol traits, PIP > 0.8**

```{r}
finemap_sig <- subset(finemap_res, group != "SNP" & susie_pip > 0.8 & !is.na(cs))

```

There were 19 significant molecular traits: 12 expression, 7 splicing.


**Compute gene PIPs:**

```{r}
combined_pip_by_context <- combine_gene_pips(susie_alpha_res, 
                                             group_by = "molecular_id",
                                             by = "context",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = TRUE)

combined_pip_by_context_sig <- subset(combined_pip_by_context, combined_pip > 0.8)

DT::datatable(combined_pip_by_context_sig,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Significant genes'),options = list(pageLength = 10) )

```

**Add gene annotations:(make sure the genome build match)** 

```{r}
ens_db <- EnsDb.Hsapiens.v75
finemap_gene_res <- subset(finemap_res, group != "SNP")
gene_ids <- unique(finemap_gene_res$molecular_id)
gene_annot <- get_gene_annot_from_ens_db(ens_db, gene_ids)
colnames(gene_annot)[colnames(gene_annot) == "gene_id"] <- "molecular_id"
head(gene_annot)

load("res.RData")
snp_map <- res$snp_map
```

**Update finemapping results with gene annotation:** 

```{r}
# Update finemap results with gene annot
finemap_res <- anno_finemap_res(finemap_res,
                                snp_map = snp_map,
                                mapping_table = gene_annot,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")

finemap_res_sig <- subset(finemap_res, group != "SNP" & gene_type == "protein_coding" & susie_pip > 0.8 & !is.na(cs))

DT::datatable(finemap_res_sig,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Significant protein-coding genes'),options = list(pageLength = 10) )

```


Compute gene PIPs across different types of molecular traits. Here, "group_by = gene_name", because molecular traits are mapped to their corresponding genes by the “gene_name” column.

```{r}
susie_alpha_res <- anno_susie_alpha_res(susie_alpha_res,
                                        mapping_table = gene_annot,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type <- combine_gene_pips(susie_alpha_res,
                                          group_by = "gene_name",
                                          by = "type",
                                          method = "combine_cs",
                                          filter_cs = TRUE,
                                          include_cs_id = TRUE)

combined_pip_by_type_sig <- subset(combined_pip_by_type, combined_pip > 0.8)

DT::datatable(combined_pip_by_type_sig,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Significant genes'),options = list(pageLength = 10) )
```

### Visualize by the gene plots

```{r}
load("weights.RData")
make_locusplot(finemap_res,
               region_id = "6_11790816_13209388",
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```


The top track showed -log10(p-value) of the association of variants (from CAD GWAS) and molecular traits (from the package computed z-scores) with the phenotype.

The mid track showed PIPs of variants and molecular traits. We limited PIP results to credible sets in the PIP track (filter_cs = TRUE). 

The next track showed the QTLs of the focal gene. We chose the molecular trait with the highest PIP (“PHACTR1” in this case). 

The bottom is the gene track. We limited results to protein coding genes by default. We could draw a red dotted line to show the PIP cutoff of 0.8 by setting highlight_pip = 0.8.


As conclusion, we found PHACTR1 as the likely causal gene of CAD. The protein encoded by this gene is a member of the phosphatase and actin regulator family of proteins. Polymorphisms in this gene are associated with susceptibility to myocardial infarction, coronary artery disease and cervical artery dissection. Alternative splicing of this gene results in multiple transcript variants. The detailed gene function and summary of this gene can be found at [genecard](https://www.genecards.org/cgi-bin/carddisp.pl?gene=PHACTR1) 

