---
title: "CAD cTWAS with LD"
author: "Xizhi Xu"
date: "02/17/2024"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

**Overview**

In this analysis, I extended to LD version of cTWAS, which means possible to find multiple causal signals per LD region. Our expectation was to find more significant genes than before. The analyzed disease was CAD, and the prediction model was trained on GTEx v8 in the PredictDB format.


## Set up

Load the required packages. 

```{r setup, message=FALSE, warning=FALSE}
### Set up
setwd("/project/xinhe/xizhi/cTWAS_2024/CAD_cTWAS")
library(data.table)
library(ctwas)
library(ggplot2)
library(workflowr)
library(EnsDb.Hsapiens.v86)

```


## Prepare the input: GWAS summary statistics, Reference (region info, SNP info, LD info), Prediction model (weights)

### GWAS Sum Stats 

The CAD summary statistics data I used was accessed from website: https://www.nature.com/articles/s41588-022-01233-6 . I filtered the original data to maintain the MAF between (0.01, 0.99). The sample size was 1165690.

```{r}
## GWAS z scores (which is the same as no LD version). Build: GRCh37. 
load("z_snp.RData")
gwas_n <- 1165690

```


### Reference

The reference data included genomic region definitions and LD reference. 
Another thing to notice: genome build needs to be matched. (Here: GRCh38 / hg38)

```{r}
## Reference: genome build 38, needs to match prediction models! 
# Region def (which is the same as no LD version)
region_file <- system.file("extdata/ldetect", "EUR.b38.ldetect.regions.RDS", package = "ctwas")
region_info <- readRDS(region_file)

# LD reference
LD_dir <- "/project2/mstephens/wcrouse/UKB_LDR_0.1"
genome_version <- "b38"

LD_filestem <- sprintf("ukb_%s_0.1_chr%s.R_snp.%s_%s", genome_version, region_info$chrom, 
                       region_info$start, region_info$stop)

region_metatable <- region_info
region_metatable$LD_file <- file.path(LD_dir, paste0(LD_filestem, ".RDS"))
region_metatable$SNP_file <- file.path(LD_dir, paste0(LD_filestem, ".Rvar"))


res <- create_snp_LD_map(region_metatable)
region_info <- res$region_info
snp_map <- res$snp_map
LD_map <- res$LD_map
```


### Prediction model

The prediction model was trained on GTEx in the PredictDB format. The molecular traits were expression, splicing and stability. Here, I included all tissues related to artery: Aorta, Coronary, Tibial.


The number of each group:


```{r}
load("weights_group_LD.RData")
print(weights_group_LD)


```



```{r, eval=FALSE}
## Prediction models: all artery tissues with eqtl, sqtl, stability-qtl
# Artery_Aorta
artery_aorta_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Aorta.db"
artery_aorta_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Aorta.db"
artery_aorta_stabqtl <- "/project2/xinhe/shared_data/multigroup_ctwas/weights/Munro_alt_stability_PredictDB/Artery_Aorta.db"

# Artery_Coronary
artery_coronary_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Coronary.db"
artery_coronary_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Coronary.db"
artery_coronary_stabqtl <- "/project2/xinhe/shared_data/multigroup_ctwas/weights/Munro_alt_stability_PredictDB/Artery_Coronary.db"

# Artery_Tibial
artery_tibial_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Tibial.db"
artery_tibial_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Tibial.db"
artery_tibial_stabqtl <- "/project2/xinhe/shared_data/multigroup_ctwas/weights/Munro_alt_stability_PredictDB/Artery_Tibial.db"

```


### Harmonization

#### Harmonize GWAS summary statistics and reference


The goal of this step was to exclude multi-allelic and strand-ambiguous variants, and make the three input data all consistent with each other. 



```{r}
## Harmonize GWAS sum stats
z_snp <- preprocess_z_snp(z_snp, snp_map, 
                          drop_multiallelic = TRUE, 
                          drop_strand_ambig = TRUE,
                          varID_converter_fun = convert_to_ukb_varIDs)

```


#### Harmonize prediction models

```{r}
# Artery_Aorta
# artery_aorta_eqtl_weights <- preprocess_weights(artery_aorta_eqtl,
#                                     region_info,
#                                     gwas_snp_ids = z_snp$id,
#                                     snp_map = snp_map,
#                                     type = "expression",
#                                     context = "Aorta",
#                                     weight_name = "Artery_expression",
#                                     weight_format = "PredictDB",
#                                     drop_strand_ambig = TRUE,
#                                     scale_predictdb_weights = TRUE,
#                                     load_predictdb_LD = TRUE,
#                                     filter_protein_coding_genes = TRUE,
#                                     varID_converter_fun = convert_to_ukb_varIDs,
#                                     ncore = 6)
# 
# artery_aorta_sqtl_weights <- preprocess_weights(artery_aorta_sqtl,
#                                                 region_info,
#                                                 gwas_snp_ids = z_snp$id,
#                                                 snp_map = snp_map,
#                                                 type = "splicing",
#                                                 context = "Aorta",
#                                                 weight_name = "Artery_splicing",
#                                                 weight_format = "PredictDB",
#                                                 drop_strand_ambig = TRUE,
#                                                 scale_predictdb_weights = TRUE,
#                                                 load_predictdb_LD = TRUE,
#                                                 filter_protein_coding_genes = TRUE,
#                                                 varID_converter_fun = convert_to_ukb_varIDs,
#                                                 ncore = 6)
# 
# artery_aorta_stabqtl_weights <- preprocess_weights(artery_aorta_stabqtl,
#                                                 region_info,
#                                                 gwas_snp_ids = z_snp$id,
#                                                 snp_map = snp_map,
#                                                 LD_map = LD_map,
#                                                 type = "stability",
#                                                 context = "Aorta",
#                                                 weight_name = "Artery_stability",
#                                                 weight_format = "PredictDB",
#                                                 drop_strand_ambig = TRUE,
#                                                 scale_predictdb_weights = FALSE,
#                                                 load_predictdb_LD = FALSE,
#                                                 filter_protein_coding_genes = TRUE,
#                                                 varID_converter_fun = convert_to_ukb_varIDs,
#                                                 ncore = 6)
# 
# aorta_weights <- c(artery_aorta_eqtl_weights, artery_aorta_sqtl_weights, artery_aorta_stabqtl_weights)
# save(aorta_weights, file = "aorta_weights.RData")
load("aorta_weights.RData")


# # Artery_Coronary
# artery_coronary_eqtl_weights <- preprocess_weights(artery_coronary_eqtl,
#                                     region_info,
#                                     gwas_snp_ids = z_snp$id,
#                                     snp_map = snp_map,
#                                     type = "expression",
#                                     context = "Coronary",
#                                     weight_name = "Coronary_expression",
#                                     weight_format = "PredictDB",
#                                     drop_strand_ambig = TRUE,
#                                     scale_predictdb_weights = TRUE,
#                                     load_predictdb_LD = TRUE,
#                                     filter_protein_coding_genes = TRUE,
#                                     varID_converter_fun = convert_to_ukb_varIDs,
#                                     ncore = 6)
# 
# artery_coronary_sqtl_weights <- preprocess_weights(artery_coronary_sqtl,
#                                                 region_info,
#                                                 gwas_snp_ids = z_snp$id,
#                                                 snp_map = snp_map,
#                                                 type = "splicing",
#                                                 context = "Coronary",
#                                                 weight_name = "Coronary_splicing",
#                                                 weight_format = "PredictDB",
#                                                 drop_strand_ambig = TRUE,
#                                                 scale_predictdb_weights = TRUE,
#                                                 load_predictdb_LD = TRUE,
#                                                 filter_protein_coding_genes = TRUE,
#                                                 varID_converter_fun = convert_to_ukb_varIDs,
#                                                 ncore = 6)
# 
# artery_coronary_stabqtl_weights <- preprocess_weights(artery_coronary_stabqtl,
#                                                 region_info,
#                                                 gwas_snp_ids = z_snp$id,
#                                                 snp_map = snp_map,
#                                                 LD_map = LD_map,
#                                                 type = "stability",
#                                                 context = "Coronary",
#                                                 weight_name = "Coronary_stability",
#                                                 weight_format = "PredictDB",
#                                                 drop_strand_ambig = TRUE,
#                                                 scale_predictdb_weights = FALSE,
#                                                 load_predictdb_LD = FALSE,
#                                                 filter_protein_coding_genes = TRUE,
#                                                 varID_converter_fun = convert_to_ukb_varIDs,
#                                                 ncore = 6)
# 
# coronary_weights <- c(artery_coronary_eqtl_weights, artery_coronary_sqtl_weights, artery_coronary_stabqtl_weights)
# save(coronary_weights, file = "coronary_weights.RData")
load("coronary_weights.RData")

# # Artery_Tibial
# artery_tibial_eqtl_weights <- preprocess_weights(artery_tibial_eqtl,
#                                                  region_info,
#                                                  gwas_snp_ids = z_snp$id,
#                                                  snp_map = snp_map,
#                                                  type = "expression",
#                                                  context = "Tibial",
#                                                  weight_name = "Tibial_expression",
#                                                  weight_format = "PredictDB",
#                                                  drop_strand_ambig = TRUE,
#                                                  scale_predictdb_weights = TRUE,
#                                                  load_predictdb_LD = TRUE,
#                                                  filter_protein_coding_genes = TRUE,
#                                                  varID_converter_fun = convert_to_ukb_varIDs,
#                                                  ncore = 6)
# 
# artery_tibial_sqtl_weights <- preprocess_weights(artery_tibial_sqtl,
#                                                  region_info,
#                                                  gwas_snp_ids = z_snp$id,
#                                                  snp_map = snp_map,
#                                                  type = "splicing",
#                                                  context = "Tibial",
#                                                  weight_name = "Tibial_splicing",
#                                                  weight_format = "PredictDB",
#                                                  drop_strand_ambig = TRUE,
#                                                  scale_predictdb_weights = TRUE,
#                                                  load_predictdb_LD = TRUE,
#                                                  filter_protein_coding_genes = TRUE,
#                                                  varID_converter_fun = convert_to_ukb_varIDs,
#                                                  ncore = 6)
# 
# artery_tibial_stabqtl_weights <- preprocess_weights(artery_tibial_stabqtl,
#                                                     region_info,
#                                                     gwas_snp_ids = z_snp$id,
#                                                     snp_map = snp_map,
#                                                     LD_map = LD_map,
#                                                     type = "stability",
#                                                     context = "Tibial",
#                                                     weight_name = "Tibial_stability",
#                                                     weight_format = "PredictDB",
#                                                     drop_strand_ambig = TRUE,
#                                                     scale_predictdb_weights = FALSE,
#                                                     load_predictdb_LD = FALSE,
#                                                     filter_protein_coding_genes = TRUE,
#                                                     varID_converter_fun = convert_to_ukb_varIDs,
#                                                     ncore = 6)
# 
# tibial_weights <- c(artery_tibial_eqtl_weights, artery_tibial_sqtl_weights, artery_tibial_stabqtl_weights)
# save(tibial_weights, file = "tibial_weights.RData")
load("tibial_weights.RData")

overall_weights <- c(aorta_weights, coronary_weights, tibial_weights)

```





## Run cTWAS

After all the input files ready, run the cTWAS.

```{r}
# ctwas_res_LD <- ctwas_sumstats(z_snp,
#                             overall_weights,
#                             region_info,
#                             LD_map,
#                             snp_map,
#                             thin = 0.1,
#                             maxSNP = 20000,
#                             min_group_size = 100,
#                             group_prior_var_structure = "shared_type",
#                             filter_L = TRUE,
#                             filter_nonSNP_PIP = FALSE,
#                             min_nonSNP_PIP = 0.5,
#                             min_abs_corr = 0.1,
#                             ncore = 6,
#                             ncore_LD = 4,
#                             save_cor = TRUE,
#                             cor_dir = "./cor_matrix",
#                             force_compute_cor = FALSE)

# save(ctwas_res_LD, file = "ctwas_res_LD.RData")
load("ctwas_res_LD.RData")
z_gene <- ctwas_res_LD$z_gene
param <- ctwas_res_LD$param
finemap_res <- ctwas_res_LD$finemap_res
susie_alpha_res <- ctwas_res_LD$susie_alpha_res
boundary_genes <- ctwas_res_LD$boundary_genes
region_data <- ctwas_res_LD$region_data
screen_res <- ctwas_res_LD$screen_res
```


In LD version, it's recommended to have a post-processing step before summarizing the results. As there are likely cross-region and LD mismatch issues, it may lead to false positive discoveries. Thus, we took extra actions to post-processing results below. 


## Post-processing results


### Cross-region LD issue


When the variants in prediction model span two regions, it would be unclear to cTWAS what region the gene should assign to. It would cause cross-region LD issue and may lead to false positive findings. 

Our solution was to perform “region merging” as a post-processing step. If any gene has variants in the weights that span two or more regions (“cross-boundary”), those regions will be merged, and cTWAS will rerun the fine-mapping step on the merged regions.


```{r, warning=FALSE}
high_PIP_finemap_gene_res <- subset(finemap_res, group != "SNP" & susie_pip > 0.5 & !is.na(cs))
high_PIP_genes <- unique(high_PIP_finemap_gene_res$id)
selected_boundary_genes <- boundary_genes[boundary_genes$id %in% high_PIP_genes, , drop=FALSE]

merge_region_res <- merge_region_data(selected_boundary_genes,
                                      region_data,
                                      region_info = region_info,
                                      LD_map = LD_map,
                                      snp_map = snp_map,
                                      weights = overall_weights,
                                      z_snp = z_snp,
                                      z_gene = z_gene,
                                      estimate_L = TRUE,
                                      maxSNP = 20000)
merged_region_data <- merge_region_res$merged_region_data
merged_region_info <- merge_region_res$merged_region_info
merged_LD_map <- merge_region_res$merged_LD_map
merged_snp_map <- merge_region_res$merged_snp_map
merged_region_id_map <- merge_region_res$merged_region_id_map
merged_region_L <- merge_region_res$merged_region_L


## Run finemapping again for merged regions
# finemap_merged_regions_res <- finemap_regions(merged_region_data,
#                                               LD_map = merged_LD_map,
#                                               weights = overall_weights,
#                                               group_prior = ctwas_res_LD$param$group_prior,
#                                               group_prior_var = ctwas_res_LD$param$group_prior_var,
#                                               L = merged_region_L,
#                                               save_cor = TRUE,
#                                               cor_dir = "./cor_matrix")
# save(finemap_merged_regions_res, file = "finemap_merged_region_res.RData")
load("finemap_merged_region_res.RData")

merged_region_finemap_res <- finemap_merged_regions_res$finemap_res
merged_region_susie_alpha_res <- finemap_merged_regions_res$susie_alpha_res

## Update finemapping results after region merge
res <- update_merged_region_finemap_res(finemap_res,
                                        susie_alpha_res,
                                        merged_region_finemap_res,
                                        merged_region_susie_alpha_res,
                                        merged_region_id_map)
updated_merged_finemap_res <- res$finemap_res
updated_merged_susie_alpha_res <- res$susie_alpha_res

screened_region_L <- screen_res$screened_region_L
updated_data_res <- update_merged_region_data(region_data, merged_region_data,
                                              region_info, merged_region_info,
                                              LD_map, merged_LD_map,
                                              snp_map, merged_snp_map,
                                              screened_region_L, merged_region_L,
                                              merged_region_id_map)
updated_region_data <- updated_data_res$updated_region_data
updated_region_info <- updated_data_res$updated_region_info
updated_LD_map <- updated_data_res$updated_LD_map
updated_snp_map <- updated_data_res$updated_snp_map
updated_region_L <- updated_data_res$updated_region_L


# z_gene and param remain unchanged (since prior parameters)

```



### LD mismatch Issue

LD mismatch between GWAS data (in-sample LD) and the reference LD could lead to false positives in fine-mapping. Because it is very time consuming to run the LD mismatch diagnosis for all the regions across the genome, we performed LD mismatch diagnosis and adjustment only for selected regions with high PIP signals in the post-processing. 


```{r}
nonSNP_PIPs <- compute_region_nonSNP_PIPs(updated_merged_finemap_res, filter_cs = TRUE)
selected_region_ids <- names(nonSNP_PIPs)[nonSNP_PIPs > 0.8]

# res_mismatch <- diagnose_LD_mismatch_susie(region_ids = selected_region_ids,
#                                   z_snp = z_snp,
#                                   LD_map = updated_LD_map,
#                                   gwas_n = gwas_n,
#                                   p_diff_thresh = 5e-8)
# save(res_mismatch, file = "res_mismatch.RData")
load("res_mismatch.RData") # After cross-region


```



```{r}
problematic_snps <- res_mismatch$problematic_snps
condz_stats <- res_mismatch$condz_stats
flipped_snps <- res_mismatch$flipped_snps

problematic_genes <- get_problematic_genes(problematic_snps,
                                           overall_weights,
                                           updated_merged_finemap_res,
                                           pip_thresh = 0.5)

```


Our basic strategy of dealing with LD mismatch was: we use the list of problematic SNPs to identify the molecular traits whose results may be affected by LD-mismatch. We would then run SuSiE fine-mapping with L = 1 in the regions containing these molecular traits, assuming a single causal signal in such a region. The fine-mapping results in this setting would be independent of LD.

We chose the genes with some plausibility of being risk genes (gene PIP > 0.5) and problematic SNPs in their weights. We would then select regions containing these problematic genes.



```{r}
problematic_region_ids <- unique(updated_merged_finemap_res[updated_merged_finemap_res$id %in% problematic_genes, "region_id"])

if (length(problematic_region_ids) > 0) {
  rerun_region_data <- updated_region_data[problematic_region_ids]
  res <- finemap_regions_noLD(rerun_region_data,
                              group_prior = ctwas_res_LD$param$group_prior,
                              group_prior_var = ctwas_res_LD$param$group_prior_var)
  rerun_finemap_res <- res$finemap_res
  rerun_susie_alpha_res <- res$susie_alpha_res
}
# 
res <- update_finemap_res(updated_merged_finemap_res,
                          updated_merged_susie_alpha_res,
                          rerun_finemap_res,
                          rerun_susie_alpha_res,
                          updated_region_ids = problematic_region_ids)
updated_finemap_res <- res$finemap_res
updated_susie_alpha_res <- res$susie_alpha_res


```


```{r, include=FALSE}
# One more step
updated_susie_alpha_res <- updated_susie_alpha_res[abs(updated_susie_alpha_res$z) > 3.2, ]
updated_finemap_res <- updated_finemap_res[abs(updated_finemap_res$z) > 3, ]

```


## Summarize and visualize results

### View the parameters and create converge plots

```{r}
make_convergence_plots(param, gwas_n, colors = c("#E69F00", "#56B4E9", "#009E73",
                                                 "#F0E442", "#0072B2", "#D55E00",
                                                 "#CC79A7", "#999999", "#E1A3FF",
                                                 "#3C3C3C"))

ctwas_parameters <- summarize_param(param, gwas_n)
ctwas_parameters

```


### Create the pie chart to see the percentage of heritability

```{r}
data <- data.frame(
  category = names(ctwas_parameters$prop_heritability),
  percentage = ctwas_parameters$prop_heritability
)

data$percentage_label <- paste0(round(data$percentage * 100, 1), "%")

ggplot(data, aes(x = "", y = percentage, fill = category)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +  # Remove background and axes
  geom_text(aes(label = percentage_label),
            position = position_stack(vjust = 0.5), size = 5) +
  scale_fill_manual(values = c("#FF9999", "#66B2FF", "#E1A3FF", "#FFCC99", "#CCCCFF", "#99CCCC", "#99FF99",
                               "#E69F00", "#999999", "#009E73")) +
  labs(fill = "Category") +
  ggtitle("Percent of heritability")

```


### Create a diagnostic plot


For all genes analyzed, we compared the Z-scores and fine-mapping PIPs. We generally expect high PIP molecular traits to have high Z-scores.


```{r}
# ## Create a diagnostic plot
ggplot(data = updated_finemap_res[updated_finemap_res$type!="SNP", ], aes(x = abs(z), y = susie_pip)) +
  geom_point() +
  labs(x = "abs(z-scores)", y = "PIPs") +
  theme_minimal()

```


### Add gene annotations (make sure the genome build match and mapping table correct) 

```{r}
mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_munro <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS")
mapping_two <- rbind(mapping_table,mapping_munro)
updated_finemap_res <- anno_finemap_res(updated_finemap_res,
                                snp_map = updated_snp_map,
                                mapping_table = mapping_two,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")

finemap_res_show <- updated_finemap_res[!is.na(updated_finemap_res$cs) & updated_finemap_res$type !="SNP",]

DT::datatable(finemap_res_show,
              caption = htmltools::tags$caption(
                style = 'caption-side: topleft; text-align = left; color:black;',
                'The annotated fine-mapping results, ones within credible sets are shown'),
              options = list(pageLength = 10))

```


### Compute gene PIPs across different types of molecular traits 

We focused on genes with combined PIP > 0.8.

```{r}
updated_susie_alpha_res <- anno_susie_alpha_res(updated_susie_alpha_res,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)
# 
combined_pip_by_type <- combine_gene_pips(updated_susie_alpha_res,
                                          group_by = "gene_name",
                                          by = "type",
                                          method = "combine_cs",
                                          filter_cs = TRUE,
                                          include_cs_id = TRUE)

combined_pip_by_type_sig <- subset(combined_pip_by_type, combined_pip > 0.8)
DT::datatable(combined_pip_by_type_sig,caption = htmltools::tags$caption(style = 'caption-side: topleft; text-align = left; color:black;','Gene PIPs, only genes with credible sets are shown'),
              options = list(pageLength = 10))


```


### Visualize by the gene locus plots

Now we created locus plot for the top genes. Since there were lots of genes with PIP = 1, we chose the 3 novel genes, and also appeared in both with LD and without LD results. 



```{r}
ens_db <- EnsDb.Hsapiens.v86

make_locusplot(updated_finemap_res,
               region_id = "2_28375910_28994693",
               ens_db = ens_db,
               weights = overall_weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")


make_locusplot(updated_finemap_res,
               region_id = "17_36141651_38653091",
               ens_db = ens_db,
               weights = overall_weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")


make_locusplot(updated_finemap_res,
               region_id = "8_143155464_145078481",
               ens_db = ens_db,
               weights = overall_weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")



```


The top track showed -log10(p-value) of the association of variants (from CAD GWAS) and molecular traits (from the package computed z-scores) with the phenotype.

The mid track showed PIPs of variants and molecular traits. We limited PIP results to credible sets in the PIP track (filter_cs = TRUE).

The next track showed the QTLs of the focal gene. We chose the molecular trait with the top 3 highest PIP (TRMT61B, ZNHIT3, BOP1).

The bottom was the gene track. We limited results to protein coding genes by default. We could draw a red dotted line to show the PIP cutoff of 0.8 by setting highlight_pip = 0.8.





### Gene set enrichment analysis


At last, we did GO enrichment analysis using the genes with PIP > 0.8.



```{r}
library(enrichR)
dbs <- c("GO_Biological_Process_2023", "GO_Cellular_Component_2023", "GO_Molecular_Function_2023")
genes <- combined_pip_by_type$gene_name[combined_pip_by_type$combined_pip > 0.8]
sprintf("The number of genes used in enrichment analysis = %s", length(genes))
GO_enrichment <- enrichr(genes, dbs)


```


Biological process: we found one GO term "Negative regulation of cell adhesion".


```{r}
db1 <- "GO_Biological_Process_2023"
df1 <- GO_enrichment[[db1]]
print(plotEnrich(GO_enrichment[[db1]]))
df1 <- df1[df1$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")]
print(df1)


```


Cellular component: we didn't find any GO terms. 


```{r}
db2 <- "GO_Cellular_Component_2023"
df2 <- GO_enrichment[[db2]]
print(plotEnrich(GO_enrichment[[db2]]))
df2 <- df2[df2$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")]
print(df2)


```


Molecular function: we didn't find any GO terms.


```{r}
db3 <- "GO_Molecular_Function_2023"
df3 <- GO_enrichment[[db3]]
print(plotEnrich(GO_enrichment[[db3]]))

df3 <- df3[df3$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")]

```


To sum up, "Negative regulation of cell adhesion" refers to biological processes that reduce, inhibit, or prevent cells from adhering (sticking) to each other or to the extracellular matrix (ECM). Negative regulation of cell adhesion can be relevant to CAD because cell adhesion plays a crucial role in vascular health. Some key connections include:

* Endothelial Dysfunction & Atherosclerosis: In CAD, endothelial cells (which line blood vessels) lose their ability to properly regulate adhesion molecules, leading to inflammation.

* Leukocyte (White Blood Cell) Adhesion & Inflammation: If adhesion is negatively regulated, immune cells might not be able to properly attach and clear damaged endothelial cells, leading to chronic inflammation—a hallmark of CAD.







