---
title: "CAD cTWAS (no LD)"
author: "Xizhi Xu"
date: "10/21/2024"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

**Version Notes**

This is the no LD version of cTWAS. The prediction model includes tissues "Aorta", "Coronary", "Tibial" with their expression and splicing. The genome build of prediction model and reference was HG38. 


## Set up

Load the required packages. 

```{r setup, message=FALSE, warning=FALSE}
### Set up
# setwd("xizhi_xin_midway3/cTWAS_2024/CAD_cTWAS")
library(data.table)
library(ctwas)
library(RSQLite)
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(workflowr)
library(enrichR)
```


## Prepare the input: GWAS summary statistics, Reference (region info, SNP info), Prediction model (weights)

### GWAS Sum Stats 

The CAD summary statistics data I used was accessed from website: https://www.nature.com/articles/s41588-022-01233-6 . I filtered the original data to maintain the MAF between (0.01, 0.99). The sample size was 1165690.

```{r, eval=FALSE}
gwas_data <- fread("GCST90132314_buildGRCh37.tsv", sep = "\t")

# Filter to remain the variants with MAF (0.01, 0.99) 
gwas_filter <- gwas_data[gwas_data$effect_allele_frequency > 0.01, ]
gwas_filter <- gwas_filter[gwas_filter$effect_allele_frequency < 0.99, ]
gwas_filter$z <- gwas_filter$beta / gwas_filter$standard_error

gwas_map <- gwas_filter[, c("chromosome", "base_pair_location", "z")]

gwas_n <- 1165690

```


### Reference

Here I got the region info from the package cTWAS. SNP info was also accessed from reference panel of cTWAS website.
Another thing to notice: genome build needs to be matched. (Here: GRCh38 / hg38)

```{r, eval=FALSE}
# Region info
region_file <- system.file("extdata/ldetect", "EUR.b38.ldetect.regions.RDS", package = "ctwas")
region_info <- readRDS(region_file)

# SNP info
ref_snp_info <- fread("ukb_b38_0.1_var_info.Rvar", sep = "\t") # 9309375
class(ref_snp_info) <- "data.frame"

res <- create_snp_map(region_info, ref_snp_info, ncore = 6)
region_info <- res$region_info
snp_map <- res$snp_map

head(region_info)

ref_data <- read.table("ukb_b37_0.1_var_info.Rvar", header = TRUE, sep = "\t")

colnames(gwas_map)[colnames(gwas_map) == "base_pair_location"] <- "pos"
colnames(gwas_map)[colnames(gwas_map) == "chromosome"] <- "chrom"
head(gwas_map)

# Merge to find rsID based on positions
z_snp <- merge(gwas_map, ref_data, by = c("pos", "chrom")) # 9155485
z_snp <- z_snp[, c("id", "alt", "ref", "z")]
colnames(z_snp)[colnames(z_snp) == "alt"] <- "A1"
colnames(z_snp)[colnames(z_snp) == "ref"] <- "A2"
head(z_snp)
```


Harmonize GWAS summary statistics:

```{r}
# Harmonize GWAS summary statistics
load("z_snp.RData")
load("res.RData")
region_info <- res$region_info
snp_map <- res$snp_map
z_snp <- preprocess_z_snp(z_snp, snp_map)
dim(z_snp)

```

### Prediction model

The prediction model was accessed from GTEx. The molecular traits were expression and splicing. Here, I included all tissues related to artery: Aorta, Coronary, Tibial.

The number of each group:

```{r}
load("weights_group.RData")
print(weights_group)

```

```{r,eval=FALSE}
# Artery_Aorta
artery_aorta_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Aorta.db"
artery_aorta_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Aorta.db"

artery_aorta_eqtl_weights <- preprocess_weights(artery_aorta_eqtl,
                                    region_info,
                                    z_snp$id,
                                    snp_map,
                                    type = "expression",
                                    context = "Aorta")

artery_aorta_sqtl_weights <- preprocess_weights(artery_aorta_sqtl,
                                          region_info,
                                          z_snp$id,
                                          snp_map,
                                          type = "splicing",
                                          context = "Aorta")


# Artery_Coronary
artery_coronary_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Coronary.db"
artery_coronary_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Coronary.db"

artery_coronary_eqtl_weights <- preprocess_weights(artery_coronary_eqtl,
                                                region_info,
                                                z_snp$id,
                                                snp_map,
                                                type = "expression",
                                                context = "Coronary")

artery_coronary_sqtl_weights <- preprocess_weights(artery_coronary_sqtl,
                                          region_info,
                                          z_snp$id,
                                          snp_map,
                                          type = "splicing",
                                          context = "Coronary")

# Artery_Tibial
artery_tibial_eqtl <- "mashr_eqtl/eqtl/mashr/mashr_Artery_Tibial.db"
artery_tibial_eqtl_weights <- preprocess_weights(artery_tibial_eqtl,
                                                   region_info,
                                                   z_snp$id,
                                                   snp_map,
                                                   type = "expression",
                                                   context = "Tibial")

artery_tibial_sqtl <- "mashr_sqtl/sqtl/mashr/mashr_Artery_Tibial.db"
artery_tibial_sqtl_weights <- preprocess_weights(artery_tibial_sqtl,
                                                 region_info,
                                                 z_snp$id,
                                                 snp_map,
                                                 type = "splicing",
                                                 context = "Tibial")

weights <- c(artery_aorta_eqtl_weights, artery_aorta_sqtl_weights, artery_coronary_eqtl_weights, 
             artery_coronary_sqtl_weights, artery_tibial_eqtl_weights, artery_tibial_sqtl_weights)

```


## Run cTWAS

After all the input files ready, run the cTWAS.

```{r}
# # ctwas_res <- ctwas_sumstats_noLD(z_snp,
#                                  artery,
#                                  region_info,
#                                  snp_map,
#                                  thin = 0.1, 
#                                  ncore = 6)


load("all_artery_weights.RData")
load("ctwas_res_all.RData")

z_gene <- ctwas_res$z_gene
param <- ctwas_res$param
finemap_res <- ctwas_res$finemap_res
susie_alpha_res <- ctwas_res$susie_alpha_res
boundary_genes <- ctwas_res$boundary_genes
region_data <- ctwas_res$region_data
screen_res <- ctwas_res$screen_res
```

### Explore and interpret the results

**Create the convergence plots:**

```{r}
gwas_n <- 1165690
make_convergence_plots(param, gwas_n)

```

These plots showed the estimated prior inclusion probability, prior effect size variance, enrichment and proportion of variance explained (PVE) over the iterations of parameter estimation. 


**Estimate the parameters:**

```{r}
ctwas_parameters <- summarize_param(param, gwas_n)
ctwas_parameters
```


**Create the pie chart to see the heritability:**

ctwas_parameters$prop_heritability contained the proportion of heritability mediated by molecular traits and variants, we visualized it using pie chart.

```{r}
data <- data.frame(
  category = names(ctwas_parameters$prop_heritability),
  percentage = ctwas_parameters$prop_heritability
)

data$percentage_label <- paste0(round(data$percentage * 100, 1), "%")

ggplot(data, aes(x = "", y = percentage, fill = category)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +  # Remove background and axes
  geom_text(aes(label = percentage_label), 
            position = position_stack(vjust = 0.5), size = 5) +
  scale_fill_manual(values = c("#FF9999", "#66B2FF", "#99FF99", "#FFCC99", "#CCCCFF", "#99CCCC", "#FF6666")) + 
  labs(fill = "Category") +
  ggtitle("Percent of heritability")
```


**Create a diagnostic plot:**

For all genes analyzed, we compared the z-scores and fine-mapping PIPs. We generally expect high PIP molecular traits to have high Z-scores as well. 

```{r}
ggplot(data = finemap_res[finemap_res$type!="SNP",], aes(x = abs(z), y = susie_pip)) +
  geom_point() +
  labs(x = "abs(z-scores)", y = "PIPs") +
  theme_minimal()
```

### Inspect and summarie results

**Add p-values to finemapping results:**


```{r}
finemap_res$pval <- z2p(finemap_res$z)
head(finemap_res)
```

**View the results: mol traits, PIP > 0.8**

```{r}
finemap_sig <- subset(finemap_res, group != "SNP" & susie_pip > 0.8 & !is.na(cs))
table(finemap_sig$type)

```

There were 15 significant molecular traits: 12 expression, 3 splicing.


**Compute gene PIPs:**

```{r}
combined_pip_by_context <- combine_gene_pips(susie_alpha_res, 
                                             group_by = "molecular_id",
                                             by = "context",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = TRUE)

combined_pip_by_context_sig <- subset(combined_pip_by_context, combined_pip > 0.8)

DT::datatable(combined_pip_by_context_sig,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Significant genes'),options = list(pageLength = 10) )

```

**Add gene annotations:(make sure the genome build match)** 

```{r}
ens_db <- EnsDb.Hsapiens.v86
mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
```

**Update finemapping results with gene annotation:** 

```{r}
# Update finemap results with gene annot
finemap_res <- anno_finemap_res(finemap_res,
                                snp_map = snp_map,
                                mapping_table = mapping_table,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")

finemap_res_sig <- subset(finemap_res, group != "SNP" & gene_type == "protein_coding" & susie_pip > 0.8 & !is.na(cs))

DT::datatable(finemap_res_sig,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Significant protein-coding genes'),options = list(pageLength = 10) )

```


Compute gene PIPs across different types of molecular traits. Here, "group_by = gene_name", because molecular traits are mapped to their corresponding genes by the “gene_name” column.


```{r}
susie_alpha_res <- anno_susie_alpha_res(susie_alpha_res,
                                        mapping_table = mapping_table,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type <- combine_gene_pips(susie_alpha_res,
                                          group_by = "gene_name",
                                          by = "type",
                                          method = "combine_cs",
                                          filter_cs = TRUE,
                                          include_cs_id = TRUE)

combined_pip_by_type_sig <- subset(combined_pip_by_type, combined_pip > 0.8)

DT::datatable(combined_pip_by_type_sig,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Significant genes'),options = list(pageLength = 10) )
```

### Visualize by the gene locus plots

We created locus plots for the top 3 gene regions. 

```{r}
make_locusplot(finemap_res,
               region_id = "2_28375910_28994693",
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")

make_locusplot(finemap_res,
               region_id = "6_13209156_14802693",
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")

make_locusplot(finemap_res,
               region_id = "17_36141651_38653091",
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")

```


The top track showed -log10(p-value) of the association of variants (from CAD GWAS) and molecular traits (from the package computed z-scores) with the phenotype.

The mid track showed PIPs of variants and molecular traits. We limited PIP results to credible sets in the PIP track (filter_cs = TRUE). 

The next track showed the QTLs of the focal gene. We chose the molecular trait with the top 3 highest PIP (TRMT61B, TBC1D7, ZNHIT3). 

The bottom was the gene track. We limited results to protein coding genes by default. We could draw a red dotted line to show the PIP cutoff of 0.8 by setting highlight_pip = 0.8.


As conclusion, we found TRMT61B, TBC1D7, ZNHIT3 as the likely causal genes of CAD. Previously, we only included Aorta tissue in the prediction model and we found ZNHIT3 as likely causal gene. Now we added 2 another tissues and we got more genes, ZNHIT3 was still there, which consistent with our previous findings. 


### Gene set enrichment analysis

At last, we did enrichment analysis using the genes with PIP > 0.8.

```{r}
dbs <- c("GO_Biological_Process_2023", "GO_Cellular_Component_2023", "GO_Molecular_Function_2023")
genes <- combined_pip_by_type$gene_name[combined_pip_by_type$combined_pip > 0.8]

# number of genes for gene set enrichment
sprintf("The number of genes used in enrichment analysis = %s", length(genes))

GO_enrichment <- enrichr(genes, dbs)

```


Biological process: we found one GO term "Circulatory System Development". 

```{r}
db1 <- "GO_Biological_Process_2023"
df1 <- GO_enrichment[[db1]]
print(plotEnrich(GO_enrichment[[db1]]))

df1 <- df1[df1$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")]
print(df1)
```


Cellular component: we didn't find any GO terms. 

```{r}
db2 <- "GO_Cellular_Component_2023"
df2 <- GO_enrichment[[db2]]
print(plotEnrich(GO_enrichment[[db2]]))

df2 <- df2[df2$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")]

```


Molecular function: we didn't find any GO terms.

```{r}
db3 <- "GO_Molecular_Function_2023"
df3 <- GO_enrichment[[db3]]
print(plotEnrich(GO_enrichment[[db3]]))

df3 <- df3[df3$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")]

```


To sum up, Circulatory System Development refers to a biological process that involves the formation and maturation of the circulatory system, which includes the heart, blood vessels, and the blood itself. Since my research topic, CAD, is closely related to abnormalities in the development and function of the circulatory system, including processes like blood vessel formation and heart development, this result makes sense. 




